{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pip install PyPDF2 pytesseract pdf2image\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting PyMuPDF\n",
      "  Obtaining dependency information for PyMuPDF from https://files.pythonhosted.org/packages/bf/dd/4aca1571508971375f4b1f849844a74cba8c10387d857c840d7c0c2d0db0/PyMuPDF-1.24.4-cp311-none-macosx_11_0_arm64.whl.metadata\n",
      "  Downloading PyMuPDF-1.24.4-cp311-none-macosx_11_0_arm64.whl.metadata (3.4 kB)\n",
      "Collecting PyMuPDFb==1.24.3 (from PyMuPDF)\n",
      "  Obtaining dependency information for PyMuPDFb==1.24.3 from https://files.pythonhosted.org/packages/7e/4a/27e4e2ce8f5d0ed1d1b2a1f7807f6158db1e8e547a7bf76ac462a800a4b4/PyMuPDFb-1.24.3-py3-none-macosx_11_0_arm64.whl.metadata\n",
      "  Downloading PyMuPDFb-1.24.3-py3-none-macosx_11_0_arm64.whl.metadata (1.4 kB)\n",
      "Downloading PyMuPDF-1.24.4-cp311-none-macosx_11_0_arm64.whl (3.0 MB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.0/3.0 MB\u001b[0m \u001b[31m5.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0ma \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hDownloading PyMuPDFb-1.24.3-py3-none-macosx_11_0_arm64.whl (14.9 MB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m14.9/14.9 MB\u001b[0m \u001b[31m5.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0m\n",
      "\u001b[?25hInstalling collected packages: PyMuPDFb, PyMuPDF\n",
      "Successfully installed PyMuPDF-1.24.4 PyMuPDFb-1.24.3\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install PyMuPDF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import re\n",
    "import PyPDF2\n",
    "import pytesseract\n",
    "from pdf2image import convert_from_path\n",
    "\n",
    "# 指定 Tesseract 的安装路径（根据实际安装路径修改）\n",
    "#pytesseract.pytesseract.tesseract_cmd = r'C:\\Program Files\\Tesseract-OCR\\tesseract.exe'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "pdf_path=\"/Users/a1-6/Project-helsinki/project_30years-butterfly/paper/2023.01.11.523549v1.full.pdf\"\n",
    "with open(pdf_path, 'rb') as file:\n",
    "    reader = PyPDF2.PdfReader(file)\n",
    "    text = \"\"\n",
    "    num_pages = len(reader.pages)\n",
    "    for page_num in range(num_pages):\n",
    "        page = reader.pages[page_num]\n",
    "        page_text = page.extract_text()\n",
    "        page_text1 = re.sub(r'^\\s\\d+$', ' ', page_text, flags=re.MULTILINE)\n",
    "        text += page_text1 + \"\\n\"\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " 1 1 \n",
      " 2 \n",
      "Warming-induced range expansio n effects on the diversity  3 \n",
      "and composition of the gut microbiome: 4 \n",
      "a case study with two butterflies 5 \n",
      " 6 \n",
      " 7 \n",
      " 8 \n",
      "Daniel Montoya1,2 & Maaike de Jong3,4 9 \n",
      " 10  \n",
      " 11  \n",
      "1Basque Centre for Climate Change (BC3), Edificio Sede 1, Planta 1, Parque Científico UPV- 12  \n",
      "EHU, Barrio Sarriena, Leioa, Spain 13  \n",
      "2IKERBASQUE, Basque Foundation for Science, Bilbao, Spain 14  \n",
      "3Royal Netherlands Academy of Arts and Sciences, 1011 JV Amsterdam, the Netherlands 15  \n",
      "4Netherlands eScience Centre 16  \n",
      " 17  \n",
      " 18  \n",
      "Author for correspondence: 19  \n",
      "Daniel Montoya, ( daniel.montoya@bc3research.org ) 20  \n",
      "Tel: (+34)  944 01 46 90  21  \n",
      "Basque Centre for Climate Change (BC3), Edificio Sede 1, Planta 1, Parque Científico UPV- 22  \n",
      "EHU, Barrio Sarriena, Leioa, Spain 23  \n",
      " 24  \n",
      " 25  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 2 26  \n",
      "Abstract 27  \n",
      "Species are the habitat of a diverse community of organisms, including bacteria, fungi and 28  \n",
      "viruses – the microbiome. Evidence shows that many species shift and/or expand their 29  \n",
      "geographical ranges following warming climate; yet, the response of host-associated 30  \n",
      "microbiome communities to species range shifts has received less attention, especially in 31  \n",
      "observational studies. Here, we use two British butterfly species for which detailed long-term 32  \n",
      "distributional data is available, and document for the first time a variety of effects of warming- 33  \n",
      "induced range expansion on the diversity and composition of their microbiome. Our results 34  \n",
      "show similar diversity and species-specific responses in the evenness of the gut microbial 35  \n",
      "communities in the two butterflies. However, butterflies at the expanded ranges showed (i) a 36  \n",
      "larger variability in the microbiome diversity, and (ii) a smaller core community of bacteria that 37  \n",
      "is to a large extent a subset of the bacterial core community in the original range. The 38  \n",
      "microbiome is responsible for many aspects of the host’s physiology and growth, and for 39  \n",
      "ecosystem function, so if the changes in the gut microbial communities reported here apply to 40  \n",
      "other species and taxonomic groups, the potential impact to biodiversity and functioning after 41  \n",
      "range expansion could be severe. 42  \n",
      " 43  \n",
      " 44  \n",
      " 45  \n",
      " 46  \n",
      " 47  \n",
      "Word count : Abstract: 195; Main text: 4543; References: 87; Figures: 5; Tables: 0   48  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 3Introduction 49  \n",
      "Current climate change takes place at unprecedented rates when compared to historical records 50  \n",
      "(Jones & Mann 2004), and its ecological consequences have been documented for thousands of 51  \n",
      "species worldwide (Parmesan & Yohe 2003). Among the well-documented responses to 52  \n",
      "increasing temperature are the organisms’ range shifts towards the poles and higher altitudes 53  \n",
      "(Grabherr, Gottfried & Pauli 1994; Parmesan 1996; Stange & Ayres 2001; Parmesan & Yohe 54  \n",
      "2003; Hickling et al 2006; Hoegh-Guldberg et al. 2008; Burrows et al. 2011; Chen et al 2011; 55  \n",
      "Pateman et al. 2012; Pauli et al. 2012; Rassmann et al 2014). There is ample evidence that such 56  \n",
      "range shifts can impact species diversity (Parmesan 1996, 2006), community composition 57  \n",
      "(Wahl et al 2011), body-size structure (Albouy et al 2014), phenotypic plasticity (Matesanz et 58  \n",
      "al. 2010), and species interactions (e.g. Tylianakis et al 2008; Ledger et al 2012, Memmott et al 59  \n",
      "2007; Durant et al 2007). For example, as species of different trophic levels do not necessarily 60  \n",
      "shift their ranges equally in response to climate change, established food web interactions may 61  \n",
      "become disrupted or spatially decoupled (Phillips et al 2010; Rassmann et al 2014; Gomez-Ruiz 62  \n",
      "& Lacher 2019). These asynchronous responses in space following recent climate change can 63  \n",
      "result in shifting patterns of species dominances and evenness within communities, and 64  \n",
      "potentially to the formation of non-analogue communities (Huntley 1991; Walther 2004; 65  \n",
      "Kullmann 2006; Hobbs et al 2009). Such changes can have important consequences for further 66  \n",
      "population responses to climate change and for ecosystem processes, and therefore need to be 67  \n",
      "considered in making inclusive predictions about the effects of a warming climate on 68  \n",
      "biodiversity and ecosystems (Gilman et al 2010). 69  \n",
      " 70  \n",
      "Despite the plethora of empirical studies documenting a spatial expansion in populations 71  \n",
      "following climate change, the response of host-associated microbiome communities to species 72  \n",
      "range shifts has received less attention. Most species are a habitat for a relatively diverse 73  \n",
      "community of organisms, including bacteria, fungi and viruses. These communities are of major 74  \n",
      "influence on organismal fitness and population ecology, but until recently it was nearly 75  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 4impossible to study microbial communities at a large scale due to methodological limitations. 76  \n",
      "Metagenomic sequencing has made it possible to identify the microbiome of organic samples, 77  \n",
      "and several studies have investigated how microbial communities of host species changes to 78  \n",
      "increasing temperature. For example, recent research has found climate-induced changes in the 79  \n",
      "microbiome community of several taxonomic groups, including Sphagnum  mosses (Carrell et al 80  \n",
      "2019), kelp (Qiu et al 2019), and the gut microbiota of amphibian (Fontaine et al 2018), reptile 81  \n",
      "(Bestion et al 2017) and rodent species (Li et al 2020). Collectively, these studies provide 82  \n",
      "evidence that increasing temperatures can impact the diversity and composition of host 83  \n",
      "microbiomes, with implications to the organism’s physiology and survival, and to ecosystem 84  \n",
      "function. However, while these results come mainly from experimental mesocosms, 85  \n",
      "comparative, observational studies of host-microbiome communities in original and expanded 86  \n",
      "distributions of range-shifting species are scarce (Jones et al 2018; Ramirez et al 2019). 87  \n",
      " 88  \n",
      "In this context, a fundamental question arises: Are there general patterns in the way the 89  \n",
      "microbiome responds to climate change-driven range shifts of the hosts, or are responses merely 90  \n",
      "idiosyncratic – i.e. each host microbiome responds differently? More specifically, what are the 91  \n",
      "effects of warming-induced range expansion on the diversity and composition of the host’s 92  \n",
      "microbiome? Several hypotheses may explain the effects of climate change on the microbial 93  \n",
      "communities of range-expanding hosts. Species undergoing range expansions may escape from 94  \n",
      "antagonistic interactions such as those from their natural enemies. For example, Engelkes et al. 95  \n",
      "(2008) showed that range-expanding plants were released from shoot and root enemy control, 96  \n",
      "whereas Menéndez et al. (2008) found similar parasitoid richness but lower parasitoid attack 97  \n",
      "rates in the expanded range of a butterfly relative to its original range. If this so-called enemy 98  \n",
      "release  hypothesis is in place, we would expect microbial communities in the expanded range to 99  \n",
      "be a subset of those in the original range. This hypothesis applies to antagonistic interactions; 100  \n",
      "yet, host-microbiome interactions can also be symbiotic. Conversely, range shifts may also lead 101  \n",
      "to the formation of new interactions in the species’ expanded range margins (e.g. Kullman 102  \n",
      "2006; Collinge & Ray 2009; Hobbs et al. 2009; Lurgi et al 2012). In other words, a shift in 103  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 5spatial distribution range may result in a parallel shift in species composition of microbial 104  \n",
      "communities. However, it remains largely unknown what is the balance between lost and new 105  \n",
      "species and interactions. Besides, most studies do not consider microbial communities, so we 106  \n",
      "know little about the potential changes in the diversity and composition of the host microbiome 107  \n",
      "following warming-induced range shifts of the host.  108  \n",
      " 109  \n",
      "British butterflies are excellent focal species to study the ecological consequences of range 110  \n",
      "shifts. Their ecology is exceptionally well known and they are highly sensitive, and respond 111  \n",
      "rapidly to, environmental change. This is because, as other insect groups, butterflies are 112  \n",
      "ectothermic, have high reproductive rates and short generation times, and a high ability to 113  \n",
      "disperse, which lead them to respond more quickly to climate change through range shift than 114  \n",
      "longer-lived and less mobile organisms, e.g. plants. Furthermore, detailed long-term 115  \n",
      "distributional data is available for UK butterflies, which is an invaluable source for research on 116  \n",
      "the effects of climate change on species’ ranges. In a study on European butterflies, 63% had 117  \n",
      "ranges that shifted to the north by 35-240 km during the 20th century, while only 3% have 118  \n",
      "shifted to the south (Parmesan 1999). Of the species with the northern range boundary in the 119  \n",
      "UK, some have shown a dramatic range shift in the past decades. For example, the Brown 120  \n",
      "Argus ( Aricia agestis ) has shifted its UK range margin with more than 100 km in the past 25 121  \n",
      "years, and the Comma ( Polygonia c-album ) has moved northwards 220 km during the same 122  \n",
      "period (UKBMS, www.ukbms.org). Butterflies are also important model species in food web 123  \n",
      "ecology research, and particularly butterfly-parasitoid interactions are well studied in this 124  \n",
      "context (e.g. Klapwijk & Lewis 2014). To a lesser extent, butterfly-pathogen interactions are 125  \n",
      "used to study ecological disease dynamics (e.g. Bradley & Altizer 2005). Surprisingly though, 126  \n",
      "the butterflies’ gut microbiome has been less studied than that of other insect groups (Engel & 127  \n",
      "Moran 2013; Hammer et al 2014; Yun et al 2 014; Coyte et al 2015; Nair & Agashe 2016), 128  \n",
      "which contrasts with the detailed information on butterflies’ distribution patterns. 129  \n",
      " 130  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 6Here, we investigate how warming-induced range shifts impact gut microbial communities in 131  \n",
      "the expanded range of butterflies. Using two butterfly species as case models, we compare 132  \n",
      "diversity, abundance and composition of butterflies’ gut bacterial communities in original and 133  \n",
      "shifted ranges. Specifically, we address two questions: (1) How does climate change-driven 134  \n",
      "range expansion affect the gut microbial diversity of expanding butterfly species?, and (2) What 135  \n",
      "are the effects of climate-driven range expansion on the abundance patterns and the microbiome 136  \n",
      "composition in the expanded margin of butterflies relative to the original range?  137  \n",
      " 138  \n",
      " 139  \n",
      "Material and Methods 140  \n",
      " 141  \n",
      "Butterfly species, site selection and field sampling 142  \n",
      "We used detailed data on the distribution and density of UK butterfly species from the UK 143  \n",
      "Butterfly Monitoring Scheme (UKBMS, www.ukbms.org), in the form of long-term (>30 years) 144  \n",
      "biweekly transect data for numerous sites across the UK. The UKBMS has documented recent 145  \n",
      "range shifts of several UK butterfly species in response to climate change, and allowed us to 146  \n",
      "carefully select the species and sites best suited to address our questions. We sampled two 147  \n",
      "butterfly species that have shown a severe shift of their northern range margin during the past 148  \n",
      "10-25 years: Brown Argus ( Aricia agestis ), and Speckled Wood ( Pararge aegeria ). We chose 149  \n",
      "sites where abundances of these butterflies have recurrently been high in previous years, thus 150  \n",
      "representing locations where the butterflies are already established, abundant, and can be easily 151  \n",
      "sampled. The same sites were chosen to sample ≥1 butterfly species if they coexisted. For each 152  \n",
      "butterfly, we selected sites that were spread out within the distribution range (both in original 153  \n",
      "and expanded margin) to avoid local effects on the microbiome. This protocol resulted in a total 154  \n",
      "of 14 sites (Figure 1): 8 sites for Brown argus (4 in original, 4 in expanded margin), and 6 sites 155  \n",
      "for Speckled wood (3 in original, 3 in expanded margin). Sampling was conducted in June- 156  \n",
      "September of 2014. Each site was visited several times over the flight period of each butterfly 157  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      "species to account for possible variation in microbial diversity due to environmental conditions 158 \n",
      "and age of the butterflies. The general flight times of the species are: Brown Argus: late July – 159 \n",
      "early September, Speckled Wood: July - early September. For each site, we collected 20 adult 160 \n",
      "individuals per butterfly species to analyse gut microbial diversity, and stored the samples at - 161 \n",
      "20˚C.  162 \n",
      " 163 \n",
      "Figure 1 . Map of the sampling locations across the distribution range of the two butterfly 164 \n",
      "species. Sites within the expanded range are not always located north of original ranges, 165 \n",
      "despite this is the common case. For each site, 20 adult individuals per butterfly species were 166 \n",
      "collected to analyse gut microbial diversity and composition. 167 \n",
      " 168 \n",
      " 169 \n",
      " 170 \n",
      "Metagenomics 171 \n",
      "After completion of the fieldwork, we extracted genomic DNA from the individual samples in 172 \n",
      "the laboratory. We focus on the gut microbiome because the likelihood of ‘contamination’ is 173 \n",
      "lower than that of external bacteria. We used Illumina High-seq to sequence the bacterial 16S 174 \n",
      "rRNA and fungal ITS ribosomal region of the pooled population samples to determine the 175 \n",
      "bacterial and fungal metagenome for each butterfly species and site. The primers were designed 176 \n",
      "with a population-specific barcode sequence so that amplification products from the different 177 \n",
      ". CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 8populations could be pooled together for a single library preparation and sequenced on a single 178  \n",
      "lane. Metagenomics sequencing was done at the University of Bristol’s Genomics Facility 179  \n",
      "(http://www.bristol.ac.uk/biology/genomics-facility/). The resulting short reads were assembled 180  \n",
      "into longer contigs, and the microbial taxonomy was determined by comparing the sequences 181  \n",
      "against online databases such as the European Ribosomal RNA database. This protocol allowed 182  \n",
      "us to identify the gut microbiome for each butterfly species – operational taxonomic unit (OTU) 183  \n",
      "– and each site.  184  \n",
      " 185  \n",
      "Statistical and network analysis 186  \n",
      "We calculated various metrics describing the diversity, abundance and composition of microbial 187  \n",
      "communities of butterflies in original versus  expanded ranges. All analyses were performed in 188  \n",
      "R software (R version 4.0.5; R Core Team, 2020).  189  \n",
      " 190  \n",
      "Diversity and abundance . We measured the number of OTUs (mean ± sd) and different features 191  \n",
      "of OTU abundance distributions. Changes in biological communities by anthropogenic 192  \n",
      "stressors, including climate change, not only affects the number of species, but also their 193  \n",
      "relative abundance and thereby dominance or evenness patterns in most ecosystems (Hillebrand 194  \n",
      "et al 2011). More importantly, evenness often responds more rapidly to environmental 195  \n",
      "constraints than species richness (Chapin et al 2000). Thus, the distribution and variance of 196  \n",
      "species within a host may be more important for aggregate performance of microbial 197  \n",
      "communities than the number of species. To assess how evenly OTU abundance is distributed 198  \n",
      "between original and expanded sites, we fitted Zipf models to rank-abundance distributions 199  \n",
      "(RADs). The Zipf  model  is given by the power law ar = Np1rγ, where ar is the predicted 200  \n",
      "abundance of the OTU rank r; N is the total number of individuals; p1 is the estimated 201  \n",
      "proportion of the most abundant OTU (rank 1); and γ is the estimated exponent of the power 202  \n",
      "law ( γ<0). Thus, a smaller value of | γ| indicates a more even distribution of abundances. We also 203  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 9estimated β-diversity of OTUs (Whittaker’s index) to investigate changes in OTU composition 204  \n",
      "between the butterflies’ original and expanded ranges.  205  \n",
      " 206  \n",
      "Core communities . Ecological communities, including host-associated microbial communities, 207  \n",
      "comprise different types of microorganisms according to the intimacy and repeatability of the 208  \n",
      "association with their host, forming a continuum from core through transient to opportunistic 209  \n",
      "organisms (Schmitt et al 2011; Phalnikar et al 2019). This has prompted the development of the 210  \n",
      "‘core microbiome’ concept as a way to identify organisms that are consistently found within a 211  \n",
      "particular host species, likely performing core metabolic functions (Astudillo-Garcia et al 2017; 212  \n",
      "McDevitt-Irwin et al 2017; Lurgi et al 2019). To examine the structuring of the most prevalent 213  \n",
      "interactions between butterflies and bacteria that are in close association with them, and whether 214  \n",
      "they change following range expansion between original and shifted ranges, we constructed the 215  \n",
      "butterfly core–microbiome community. The butterfly core–microbiome community was defined 216  \n",
      "as the subset of OTUs found in butterfly hosts that are (i) present in at least two-thirds of the 217  \n",
      "sites in each part of the butterfly’s range, and (ii) present with a relative abundance ≥0.001% 218  \n",
      "across the whole dataset within each part of the range (Astudillo-García et al 2017; Lurgi et al 219  \n",
      "2019). Results were relatively robust to changing abundance criteria (see Table S1 in 220  \n",
      "Supplementary Material).  221  \n",
      " 222  \n",
      " 223  \n",
      "Results 224  \n",
      " 225  \n",
      "Diversity and abundance of butterfly’s microbiome 226  \n",
      "Richness of microbial communities varies across different host species. As with most organisms 227  \n",
      "inhabiting the guts of most insect species, bacterial species comprise all or the majority of 228  \n",
      "organisms of the gut microbiome. With respect to bacterial taxonomy, our findings are 229  \n",
      "generally congruent with previous work on insect-associated bacterial communities (Engel & 230  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      "Moran 2013; Yun et al 2014; van Schooten et al 2018). The most abundant and frequent OTUs 231 \n",
      "in the studied butterflies are also common members of other insect-associated microbiomes: 232 \n",
      "Proteobacteria and Firmicutes  were the dominant phyla and represented a 50% and 21% of the 233 \n",
      "total sequences, respectively, with abundant phyla of Bacteroidetes , Actinobacteria , and 234 \n",
      "Tenericutes  in some butterfly samples (Figure S1 and Table S3 in Supplementary Material). 235 \n",
      "There was no evidence to support differences in the mean number of OTUs for butterflies at the 236 \n",
      "expanded range ( F = 1.472, p = 0.248), suggesting that range expansion did not result in a net 237 \n",
      "gain or loss of OTUs. Yet, a larger variability in the number of OTUs was observed in expanded 238 \n",
      "versus original ranges (Brown argus: {31, 55} vs {39, 53}; Speckled wood: {31, 46} vs {39, 239 \n",
      "45}) (Figure 2). 240 \n",
      " 241 \n",
      "Figure 2 . Diversity of the butterflies’ gut microbiome in original and expanded ranges. Plots 242 \n",
      "show the mean and standard deviation of the number of bacterial OTUs for per butterfly 243 \n",
      "species. There was no evidence supporting differences in mean OTU richness (F = 1.472, p = 244 \n",
      "0.248), despite a larger variability in expanded versus original range is clear. 245 \n",
      " 246 \n",
      " 247 \n",
      " 248 \n",
      " 249 \n",
      "RAD analysis shows similar and more even distribution of OTU abundances in the butterflies’ 250 \n",
      "original range for brown argus and speckled wood, respectively ( Zipf fits, mean | γ| across sites: 251 \n",
      "Brown argus = {0.605 (original), 0.6 (expanded)}, Speckled wood = {0.61 (original), 0.67 252 \n",
      ". CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      "(expanded)}). That is, whereas no difference in evenness are observed for brown argus, certain 253 \n",
      "bacteria are more dominant in the expanded margins of the speckled wood populations (Figure 254 \n",
      "3). When divided into abundance categories we find that, whereas the abundance of dominant 255 \n",
      "bacteria is similar in both original and expanded ranges, differences in the bacterial abundance 256 \n",
      "distribution mostly arise at intermediate and low levels of abundance (Figure 4). This pattern is 257 \n",
      "consistent either aggregating all sites per butterfly or considering individual sites separately 258 \n",
      "(Figure S2 in Supplementary Material). This is further supported by the β-diversity analysis of 259 \n",
      "the three abundance categories (>0.01, 0.01 – 0.001, <0.001): despite no major turnover is 260 \n",
      "observed within the most abundant bacteria, intermediate and rare bacteria show a high turnover 261 \n",
      "(Whittaker’s index for most, intermediate and least abundant OTUs: Brown argus = {0.22, 0.34, 262 \n",
      "0.75}, Speckled wood = {0.39, 0.48, 0.71}). Although these numbers can slightly vary when 263 \n",
      "using different thresholds to establish abundance categories (Table S2 in Supplementary 264 \n",
      "Material), β-diversity is always lower for the dominant bacteria and increases as bacteria 265 \n",
      "become less abundant. 266 \n",
      " 267 \n",
      "Figure 3 . Rank-abundance distribution of bacterial OTUs in original and expanded ranges.  (A) 268 \n",
      "Brown argus; (B) Speckled wood. Abundance values corresponding to all sites within each 269 \n",
      "range are contained within the shadows, whose limits are determined by the minimum and 270 \n",
      "maximum values across the range of bacteria abundances within each range (original and 271 \n",
      "expanded) 272 \n",
      " 273 \n",
      " 274 \n",
      ". CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 275 \n",
      " 276 \n",
      "Figure 4 . Rank-abundance distribution of bacterial OTUs, divided into abundance categories 277 \n",
      "(black – high, blue – intermediate, red – low), in original and expanded ranges. Upper row: 278 \n",
      "Brown argus; Bottom row: Speckled wood. Abundance categories (and their colors) are 279 \n",
      "established for bacteria within the original range and maintained in the expanded range 280 \n",
      "graphs, allowing to visualize bacterial turnover between original and expanded ranges. Most of 281 \n",
      "the differences in the abundance distribution of bacteria arise at intermediate and low levels of 282 \n",
      "abundance, whereas dominant bacteria are persistent across original and expanded ranges. 283 \n",
      "Numbers indicate β-diversity (Whittaker’s index) for the different abundance categories (higher 284 \n",
      "β indicates higher turnover). 285 \n",
      " 286 \n",
      " 287 \n",
      " 288 \n",
      "Core communities  289 \n",
      "Figure 5 shows the core communities of each butterfly species with original and expanded range 290 \n",
      "sites. Despite the application of such strict constraints on the definition of the core microbiome 291 \n",
      ". CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      "(see Material and Methods), a relatively large butterfly core–microbiome community remained. 292 \n",
      "Butterflies in the original sites have a larger core community of gut bacteria relative to that of 293 \n",
      "the expanded regions. Besides, the core community at the expanded margin is, to a large extent, 294 \n",
      "contained within the original range for the two butterflies, thus suggesting that butterflies left 295 \n",
      "some bacteria behind during climate-driven range expansion. The butterfly-microbiome also 296 \n",
      "acquires new bacteria at the expanded margin, this resulting in a similar mean number of OTUs 297 \n",
      "(Figure 2). These results were robust to changing abundance criteria (see Table S1 in 298 \n",
      "Supplementary Material): for both butterflies, at least two-thirds of the core community of 299 \n",
      "bacteria at the expanded range are a subset of the core community within the original range of 300 \n",
      "distribution. 301 \n",
      " 302 \n",
      "Figure 5 . Butterfly-bacteria core communities across distributional ranges. For each butterfly, 303 \n",
      "original and expanded range sites are included. The number of bacterial OTUs in original and 304 \n",
      "expanded range sites are indicated, as well as the number of shared bacterial OTUs across the 305 \n",
      "entire distribution range of each butterfly species. 306 \n",
      " 307 \n",
      " 308 \n",
      " 309 \n",
      " 310 \n",
      " 311 \n",
      "Discussion 312 \n",
      ". CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 14To our knowledge, this is the first observational study in real-world communities that 313  \n",
      "investigates changes in the diversity and composition of the host-microbiome in species that 314  \n",
      "have experienced warming-induced range expansion. Using two butterfly species as case 315  \n",
      "studies, our results reveal changes in butterfly microbial communities in the expanded range of 316  \n",
      "the butterflies’ distribution. Despite no significant changes in mean OTU numbers and species- 317  \n",
      "specific responses in evenness, two general patterns were observed at the shifted ranges of the 318  \n",
      "butterflies that are consistent across the two species: (i) a larger variability in the diversity of gut 319  \n",
      "bacterial communities, and (ii) a smaller core community of bacteria that is to a large extent a 320  \n",
      "subset of the bacterial core community in the original range.  321  \n",
      " 322  \n",
      "Despite our results find support for negative effects of range shifts on the average bacterial 323  \n",
      "diversity of the gut microbiome (e.g. core communities are smaller at expanded ranges), they 324  \n",
      "are compensated with the acquisition of new bacteria. The balance between lost and gained 325  \n",
      "bacteria is neutral, thus resulting in microbiomes that are not significantly different in terms of 326  \n",
      "mean OTUs despite a larger variability in their numbers. Our analysis also reveals that the core 327  \n",
      "bacterial microbiome of the butterflies becomes less diverse after range expansion. Moreover, 328  \n",
      "the core microbiome at the expanded margin is, to a large extent, contained within the original 329  \n",
      "range for both butterflies. Such reduction in the diversity of the core bacterial communities in 330  \n",
      "parallel to similar mean, but higher diversity levels, may be driven by several, non-exclusive 331  \n",
      "factors. First, butterflies acquire their microbiome from the environment via food sources 332  \n",
      "(horizontal transmission), and thus smaller core communities at the expanded ranges of the 333  \n",
      "butterflies’ distribution would result from the fact that only a fraction of the bacteria found in 334  \n",
      "the butterflies’ core gut microbiome is common in both parts of the distribution range. 335  \n",
      "However, range-shifting populations tend to have broader diets, and this may counteract the 336  \n",
      "effects of potentially lower diversity of food sources that provide core bacteria (Braschler & 337  \n",
      "Hill 2007; Pateman et al 2012; Lancaster 2016; Eloy de Amorim et al 2017; Suehiro et al 2017; 338  \n",
      "Shively et al 2018; Singer & Parmesan 2020); indeed, Lancaster (2020) analyzed 4410 butterfly 339  \n",
      "species across the globe and found that geographic range expansions are linked to butterflies’ 340  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 15diet expansion. Collectively, our results suggest that smaller core microbiomes in the expanded 341  \n",
      "range of the butterflies’ distribution are compensated by broader diet breadths, this leading to 342  \n",
      "bacterial communities that are highly variable, yet similar in their mean OTU diversity. 343  \n",
      " 344  \n",
      "An alternative explanation to the larger variability in bacterial diversity is that butterflies at the 345  \n",
      "expanded range are going through a transient state. On one hand, the gut microbiome is 346  \n",
      "horizontally transmitted and depends on both the availability of food resources and the 347  \n",
      "butterflies’ diet breadth. Diet breadth tends to increase with latitude and decline with population 348  \n",
      "age in butterflies (Singer & Parmesan 2020). Thus, the high variability of bacterial diversity 349  \n",
      "observed in the expanded ranges could reflect a transient state where the studied butterflies’ 350  \n",
      "populations are still young and not yet stabilised. Such variability could also be indicative of the 351  \n",
      "strength of host-bacterial interactions; in the case of symbiont interactions, low variability 352  \n",
      "would indicate more obligate interactions between butterflies and bacteria (Phalnikar et al 353  \n",
      "2019). On the other hand, it could also be argued that the studied butterflies have already 354  \n",
      "reached an equilibrium state in their expanded margins, as (i) butterflies exhibit a strong 355  \n",
      "response to warming given their ectothermic nature, short generation times, and high dispersal 356  \n",
      "ability, and (ii) we chose sites where the butterflies were already present in the last 10-25 years 357  \n",
      "and with high abundances in previous years before sampling. However, the studied butterflies 358  \n",
      "have 1-2 generations per year, so it is debatable whether their populations are at equilibrium. 359  \n",
      "Further, newer populations in the expanded margins are usually smaller and/or more isolated, 360  \n",
      "and this may reduce microbial exchange via the environment. Studies that include temporal 361  \n",
      "resolution will be needed to determine whether the observed reduction in microbiome diversity 362  \n",
      "reflects a transient state. 363  \n",
      " 364  \n",
      "Our results show that bacteria richness and evenness can respond independently to warming- 365  \n",
      "induced range expansion, as shown by changes in the bacterial abundance distributions of one 366  \n",
      "of the butterflies’ microbiome. Specifically, whereas no difference is reported for brown argus, 367  \n",
      "bacterial abundances in the shifted ranges of speckled wood populations are less evenly 368  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 16distributed compared to the original range. This generally agrees with recent findings 369  \n",
      "documenting that species richness and abundances can respond simultaneously to global change 370  \n",
      "factors in various and independent ways (e.g. Chapin et al 2000; Hillebrand et al 2011; 371  \n",
      "McWilliams et al 2019). Our results further show that the identity of the most dominant bacteria 372  \n",
      "in the microbiome persists in both parts of the butterflies’ range, and that the effects of range 373  \n",
      "expansion on the butterflies’ microbiome has a higher impact in bacteria with intermediate and 374  \n",
      "low levels of abundance. These are the less stable bacteria that experience the highest turnover 375  \n",
      "between original and shifted ranges. Differences in the abundance distributions and evenness 376  \n",
      "patterns of bacteria could be explained by the fact  that, in butterflies, the impact of gut microbes 377  \n",
      "on their insect hosts range along a continuum from strong to weak dependence, to no 378  \n",
      "association (Phalnikar et al 2019). Moreover, even when acquired independently each 379  \n",
      "generation, insect gut communities are not expected to be random assemblages of bacteria from 380  \n",
      "the food or local environment (most studies show differences between microbes in the guts vs 381  \n",
      "those ingested with food). Rather, gut communities seem to be dominated by widely distributed 382  \n",
      "bacteria that appear to colonize hosts opportunistically (Engel & Moran 2013). Thus, whereas 383  \n",
      "rare bacteria might be more responsive to warming and/or less widely distributed, abundant gut 384  \n",
      "bacteria become abundant because they establish intimate associations with their host butterflies 385  \n",
      "(e.g. symbiosis) and/or are present in high abundances across the distribution range of 386  \n",
      "butterflies (opportunistic).  387  \n",
      " 388  \n",
      "How do the observed changes in the microbiome diversity and composition affect butterflies’ 389  \n",
      "survival and ecosystem function? Although this is beyond the scope of this work, the results 390  \n",
      "here reported have several conservation implications. First, extinction risk decreases with 391  \n",
      "evenness, as shown by e.g. experiments where plant extinctions were higher in plots sown with 392  \n",
      "low compared to high evenness (Wilsey & Polley 2004). In the case of speckled wood 393  \n",
      "populations, this suggests that butterflies’ microbiome at the shifted ranges may lose more 394  \n",
      "microbial diversity, especially that of rare bacteria. Second, experimental studies across 395  \n",
      "different taxonomic groups show that alterations in diversity and composition of host 396  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 17microbiomes in response to warming can strongly impact the host’s physiology and survival 397  \n",
      "(Bestion et al 2017; Fontaine et al 2018; Carrel et al 2019; Qiu et al 2019; Li et al 2020). 398  \n",
      "Besides, changes in evenness are as important as changes in species richness and can have 399  \n",
      "important consequences for ecosystem function (Hillebrand et al 2011). Third, our analyses did 400  \n",
      "not discriminate between antagonistic and symbiotic bacteria, so relative changes in the 401  \n",
      "proportion of these interactions might be taking place. In contrast to the benefits associated with 402  \n",
      "the release from antagonistic interactions (e.g. predators, pathogens) on range-expanding 403  \n",
      "species (Callaway et al 2004; Mueller & Hellmann 2008; Mitchell et al 2006; Lavergne et al 404  \n",
      "2010; Hellmann et al 2012), a higher symbiont gut bacterial diversity is often beneficial to hosts 405  \n",
      "(Bolnick et al 2014). Thus, climate-driven changes in bacterial composition could be 406  \n",
      "detrimental for host butterflies, particularly if it entails a loss of essential functions (Mandrioli 407  \n",
      "2012; Le Chatelier et al 2013; Kikuchi et al 2016). 408  \n",
      " 409  \n",
      "Our study has several limitations that future analyses should overcome. For example, we used 410  \n",
      "two butterfly species as case studies. Although these butterflies were selected because of the 411  \n",
      "availability of detailed information on their range expansion, adding more species and sites 412  \n",
      "would provide a more complete picture of the effects of warming-induced range expansion and 413  \n",
      "whether the results here reported can be generalized to other butterflies and different taxonomic 414  \n",
      "groups. Further, despite each site was visited several times over the butterflies’ flight period to 415  \n",
      "account for possible variation in microbial diversity, we acknowledge that this procedure 416  \n",
      "captures within-year variation but does not account for the long-term patterns of range 417  \n",
      "expansion. Studies should incorporate a dynamical perspective with time-series to overcome 418  \n",
      "this limitation and reveal whether range-shifting butterflies are going through a transient state or 419  \n",
      "have already reached a dynamical equilibrium within their expanded ranges. Despite these 420  \n",
      "limitations, to our knowledge this is the first comparative, observational study of host- 421  \n",
      "microbiome communities in original and expanded distributions of range-shifting species, and a 422  \n",
      "useful first step to contribute to furthering our understanding of the effects of warming on the 423  \n",
      "diversity and composition and of host bacterial communities after range expansion. 424  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 18 425  \n",
      "There is ample evidence across geographical areas and taxonomic groups on species’ range 426  \n",
      "shifts towards the poles and higher altitudes as a response to warming climate. Here, we have 427  \n",
      "used two butterfly species as case studies and documented for the first time a variety of effects 428  \n",
      "of warming-induced range expansion on the diversity and composition of their gut microbial 429  \n",
      "communities. These range from non-significant changes (bacterial richness), to idiosyncratic or 430  \n",
      "butterfly-dependent changes (evenness patterns), and changes that are consistent across the two 431  \n",
      "butterflies (variability in bacterial richness, core microbiome composition). The microbiome is 432  \n",
      "responsible for many aspects of the host’s physiology and growth, and for ecosystem function, 433  \n",
      "so if the changes in the gut microbial communities reported here apply to other species and 434  \n",
      "taxonomic groups, the potential impact to biodiversity and functioning after range expansion 435  \n",
      "could be severe. Future studies should aim at investigating the degree to which the patterns 436  \n",
      "observed in this study are consistent and the consequences of its alteration to species and 437  \n",
      "ecosystems. 438  \n",
      " 439  \n",
      " 440  \n",
      "Acknowledgements 441  \n",
      "We thank Emmeline Topp, Joanne Morten, Kees Wanders, Alicia Hodson and Jon Bridle for 442  \n",
      "their assistance in conducting field and lab work, and Tom Batstone for the bioinformatic 443  \n",
      "analyses. This research was financially supported by a British Ecological Society Large 444  \n",
      "Research Grant.  445  \n",
      "  446  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 19REFERENCES 447  \n",
      " 448  \n",
      "1. Albouy, C. et al. (2013) Projected climate change and the changing biogeography of coastal 449  \n",
      "Mediterranean fishes. Journal of Biogeography 40, 534-547. 450  \n",
      "2. Astudillo-Garcia, C. et al. (2017). Evaluating the core microbiota in complex communities: a  451  \n",
      "systematic investigation. Environmental Microbiology 19, 1450-1462. 452  \n",
      "3. Bestion, E., Jacob, S., Zinger, L., Di Gesu, L., Richard, M., White, J. & Cote, J. (2017). 453  \n",
      "Climate warming reduces gut microbiota diversity in a vertebrate ectotherm. Nature Ecology 454  \n",
      "& Evolution 1(6), 0161. 455  \n",
      "4. Blüthgen, N. et al. (2006). Measuring specialization in species interaction networks. BMC 456  \n",
      "Ecology 6, 9. 457  \n",
      "5. Bolnick, D.I. et al. (2014). Individuals' diet diversity influences gut microbial diversity in 458  \n",
      "two freshwater fish (threespine stickleback and Eurasian perch). Ecology Letters 17, 979- 459  \n",
      "987. 460  \n",
      "6. Borrvall, C. et al. (2000). Biodiversity lessens the risk of cascading extinction in model food 461  \n",
      "webs. Ecology Letters 32, 131-136. 462  \n",
      "7. Bradley, C. A. & Altizer, S. (2005). Parasites hinder monarch butterfly flight: implications 463  \n",
      "for disease spread in migratory hosts. Ecology Letters 8, 290-300.  464  \n",
      "8. Braschler, B. & Hill, J. K. (2007). Role of larval host plants in the climate-driven range 465  \n",
      "expansion of the butter y Polygonia c-album . Journal of Animal Ecology  76, 415-423.  466  \n",
      "9. Burrows, M.T., Schoeman, D.S., Buckley, L.B., Moore, P., Poloczanska, E.S., Brander, 467  \n",
      "K.M. et al. (2011). The pace of shifting climate in marine and terrestrial ecosystems. Science 468  \n",
      "334, 652-655. 469  \n",
      "10. Callaway, R.M. et al. (2004). Soil biota and exotic plant invasion. Nature 427, 731-733.  470  \n",
      "11. Carrell, A.A. et al (2019). Experimental warming alters the community composition, 471  \n",
      "diversity, and N2 fixation activity of peat moss (Sphagnum fallax) microbiomes. Global 472  \n",
      "Change Biology 25, 2993-3004. 473  \n",
      "12. Chapin III, F. S. et al. (2000). Consequences of changing biodiversity. Nature 405, 234-242. 474  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 2013. Chen, I. C. et al. (2011). Rapid range shifts of species associated with high levels of climate 475  \n",
      "warming. Science 333, 1024-1026. 476  \n",
      "14. Collinge S.K. & Ray C. (2009). Transient patterns in the assembly of vernal pool plant 477  \n",
      "communities. Ecology 90, 3313-23. 478  \n",
      "15. Coyte, K.Z., Schluter, J. & Foster, K.R. (2015). The ecology of the microbiome: Networks, 479  \n",
      "competition, and stability. Science 350, 663-666. 480  \n",
      "16. Dunne, J.A. et al. (2002). Network structure and biodiversity loss in food webs: robustness 481  \n",
      "increases with connectance. Ecology Letters 5, 558-567 482  \n",
      "17. Durant, J.M., Hjermann, D.Ø., Ottersen, G. & Stenseth, N.C. (2007) Climate and the match 483  \n",
      "or mismatch between predator requirements and resource availability. Climate Research 33, 484  \n",
      "271-283.  485  \n",
      "18. Eloy de Amorim, M. et al. (2017). Lizards on newly created islands independently and 486  \n",
      "rapidly adapt in morphology and diet. Proceedings of the National Academy of Sciences 487  \n",
      "USA  114, 8812-8816.  488  \n",
      "19. Engel, P. & Moran, N.A. (2013). The gut microbiota of insects – diversity in structure and 489  \n",
      "function. FEMS Microbiology Review 37, 699-735. 490  \n",
      "20. Engelkes, T. et al. (2008). Successful range-expanding plants experience less above-ground 491  \n",
      "and below-ground enemy impact. Nature 456, 946-948. 492  \n",
      "21. Fontaine, S.S., Novarro, A.J. & Kohl, K.D. (2018). Environmental temperature alters the 493  \n",
      "digestive performance and gut microbiota of a terrestrial amphibian. Journal of Experimental 494  \n",
      "Biology 221, jeb187559. 495  \n",
      "22. Gilman, S. E. et al. (2010). A framework for community interactions under climate change. 496  \n",
      "Trends in Ecology & Evolution 25, 325-331. 497  \n",
      "23. Gomez-Ruiz, E.P. & Lacher Jr, T.E. (2019). Climate change, range shifts, and the disruption 498  \n",
      "of a pollinator-plant complex. Scientific Reports 9, 1-10. 499  \n",
      "24. Grabherr, G., Gottfried, M. & Pauli, H. (1994). Climate effects on mountain plants. Nature 500  \n",
      "369, 448.  501  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 2125. Gröbler, B.C. & O.T. Lewis. (2008). Response of native parasitoids to a range-expanding 502  \n",
      "host. Ecological Entomology 33, 453-463.  503  \n",
      "26. Hackett, T., Sauve, A., Davies, N., Montoya, D., Tylianakis, J., Memmott, J. (2019). 504  \n",
      "Reshaping our understanding of species’ roles in landscape-scale networks. Ecology Letters 505  \n",
      "22, 1367-1377. 506  \n",
      "27. Hammer, T.J., McMillan, W.O. & Fierer, N. (2014). Metamorphosis of a Butterfly- 507  \n",
      "Associated Bacterial Community. PLoS ONE 9(1), e86995. 508  \n",
      "28. Hellmann, J.J., Prior, K.M. & Pelini, S.L. (2012). The influence of species interactions on 509  \n",
      "geographic range change under climate change. Annals of the New York Academy of 510  \n",
      "Sciences 1249, 18-28. 511  \n",
      "29. Hickling R., Roy D.B., Hill J.K., Fox R., Thomas C.D. (2006). The distributions of a wide 512  \n",
      "range of taxonomic groups are expanding polewards. Global Change Biology 12, 450-455.  513  \n",
      "30. Hillebrand, H., Bennett, D.M. & Cadotte, M.A. (2008). Consequences of dominance: a 514  \n",
      "review of evenness effects on local and regional ecosystem processes. Ecology 89, 1510- 515  \n",
      "1520. 516  \n",
      "31. Hobbs, R. J., Higgs, E. & Harris, J. A. (2009). Novel ecosystems: implications for 517  \n",
      "conservation and restoration. Trends in Ecology & Evolution 24, 599-605.  518  \n",
      "32. Hoegh-Guldberg, O., Hughes, L., McIntyre, S., Lindenmayer, D.B., Parmesan, C., 519  \n",
      "Possingham, H.P. et al. (2008). Assisted colonization and rapid climate change. Science 321, 520  \n",
      "345-346. 521  \n",
      "33. Huntley, B. (1991). How plants respond to climate change— migration rates, individualism 522  \n",
      "and the consequences for plant communities. Annals of Botany 67, 15-22.  523  \n",
      "34. Jones, P. D. & Mann, M. E. (2004). Climate over past millennia. Reviews of Geophysics 42, 524  \n",
      "2. 525  \n",
      "35. Jones, J., DiBattista, J.D., Stat, M., Bunce, M., Boyce, M.C., Fairclough, D.V., Travers, M.J. 526  \n",
      "& Huggett, M.J. (2018). The Microbiome of the Gastrointestinal Tract of a Range-Shifting 527  \n",
      "Marine Herbivorous Fish. Frontiers in Microbiology 9, 1-13. 528  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 2236. Kikuchi, Y. et al. (2016). Collapse of Insect Gut Symbiosis under Simulated Climate 529  \n",
      "Change. mBio 7, e01578-16 530  \n",
      "37. Klapwijk, M.J. & Lewis, O. (2014). Spatial ecology of host-parasitoid interactions: A 531  \n",
      "threatened butterfly and its specialised parasitoid. Journal of Insect Conservation 18, 437- 532  \n",
      "445. 533  \n",
      "38. Kullmann, L. (2006) Long-term geobotanical observations of climate change impacts in the 534  \n",
      "Scandes of West-Central Sweden. Nordic Journal of Botany 24, 445-467.  535  \n",
      "39. Lancaster, L. T. (2016). Widespread range expansions shape latitudinal variation in insect 536  \n",
      "thermal limits. Nature Climate Change  6, 618-621.  537  \n",
      "40. Lancaster, L.T. (2020). Host use diversification during range shifts shapes global variation in 538  \n",
      "Lepidopteran dietary breadth. Nature Ecology & Evolution 4, 963-969. 539  \n",
      "41. Lavergne, S., Mouquet, N., Thuiller, W. & Ronce, O. (2010). Biodiversity and Climate 540  \n",
      "Change: Integrating Evolutionary and Ecological Responses of Species and Communities. 541  \n",
      "Annual Review in Ecology, Evolution & Systematics 41, 321-350. 542  \n",
      "42. Le Chatelier, E. et al. (2013). Richness of human gut microbiome correlates with metabolic 543  \n",
      "markers. Nature 500, 541-546.  544  \n",
      "43. Ledger, M. A. et al. (2013). Drought alters the structure and functioning of complex food 545  \n",
      "webs. Nature Climate Change 3, 223-227. 546  \n",
      "44. Li, G., Yin, B., Li, J., Wang, J., Wei, W., Bolnick, D.I., Wan, X., Zhu, B. & Zhang, Z. 547  \n",
      "(2020). Host-microbiota interaction helps to explain the bottom-up effects of climate change 548  \n",
      "on a small rodent species. The ISME Journal 4,  1795-1808. 549  \n",
      "45. Lurgi, M., López, B.C. & Montoya, J.M. (2012). Novel communities from climate change. 550  \n",
      "Philosophical Transactions of the Royal Society of London B 367, 2913-2922. 551  \n",
      "46. Lurgi, M., Thomas, T., Wemheuer, B., Webster, N.S. & Montoya, J.M. (2019). Modularity 552  \n",
      "and predicted functions of the global sponge-microbiome network. Nature Communications 553  \n",
      "10, 992. 554  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 2347. Macfadyen, S., Gibson, R. H., Symondson, W. O. C. & Memmot, J. (2011). Landscape 555  \n",
      "structure influences modularity patterns in farm food webs: consequences for pest control. 556  \n",
      "Ecological Applications 21, 516-524. 557  \n",
      "48. Mandrioli, M. (2012). Someone like it hot? Effects of global warming on insect immunity 558  \n",
      "and microbiota. Invertebrate Survival Journal 9, 58-63. 559  \n",
      "49. Matesanz, S., Gianoli, E. & Valladares F. (2010). Global change and the evolution of 560  \n",
      "phenotypic plasticity in plants. In: Schlichting CD, Mousseau TA (eds) Year in evolutionary 561  \n",
      "biology. Wiley-Blackwell, Malden, pp 35–55 562  \n",
      "50. McDevitt-Irwin, J. M., Baum, J. K., Garren, M. & Thurber, R. L. V. (2017). Responses of 563  \n",
      "coral-associated bacterial communities to local and global stressors. Frontiers in Marine 564  \n",
      "Science 4, 1-16.  565  \n",
      "51. McWilliam, C., Lurgi, M., Montoya, J.M. Sauve, A. & Montoya, D. (2019). The stability of 566  \n",
      "multitrophic communities under habitat loss. Nature Communications 10, 2322. 567  \n",
      "52. Memmott, J. Craze, P.G., Waser, N.M. & Price, M.V. (2007). Global warming and the 568  \n",
      "disruption of plant–pollinator interactions. Ecology Letters 10, 710-717. 569  \n",
      "53. Menéndez, R. et al. (2008). Escape from natural enemies during climate-driven range 570  \n",
      "expansion: a case study. Ecological Entomology 33, 413-421. 571  \n",
      "54. Mitchell C.E., Agrawal A.A., Bever J.D., Gilbert G.S., Hufbauer R.A., et al. (2006). Biotic 572  \n",
      "interactions and plant invasions. Ecology Letters 9, 726-40. 573  \n",
      "55. Montoya, D. (2008). Habitat loss, dispersal, and the probability of extinction of tree species. 574  \n",
      "Communicative & Integrative Biology 1(2), 146-147. 575  \n",
      "56. Montoya, D. (2021). Challenges and directions toward a general theory of ecological 576  \n",
      "recovery dynamics: A metacommunity perspective. One Earth 4(8), 1083-1094. 577  \n",
      "57. Montoya, D., Haegeman, B., Gaba, S., De Mazancourt, C. & Loreau, M. (2021). Journal of 578  \n",
      "Ecology 109(8), 2991-3006. 579  \n",
      "58. Mueller, J.M. & J.J. Hellmann. (2008). An assessment of invasion risk from assisted 580  \n",
      "migration. Conservation Biology 22, 562-567. 581  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 2459. Nair, A. & Agashe, D. (2016). Host-specific spatial and temporal variation in culturable gut 582  \n",
      "bacterial communities of dragonflies. Current Science 110(8), 1513-1523. 583  \n",
      "60. Parmesan, C. (1996). Climate and species’ range. Nature 382, 765–766.  584  \n",
      "61. Parmesan, C. (1999). Poleward shifts in geographical ranges of butterfly species associated 585  \n",
      "with regional warming. Nature 399, 579-583. 586  \n",
      "62. Parmesan, C. & Yohe, G. (2003). A globally coherent fingerprint of climate change impacts 587  \n",
      "across natural systems. Nature 421, 37-42. 588  \n",
      "63. Parmesan, C. (2006). Ecological and evolutionary responses to recent climate change. 589  \n",
      "Annual Review of Ecology, Evolution & Systematics 37, 637-669. 590  \n",
      "64. Pateman, R.M., Hill, J.K., Roy, D.B., Fox, R. & Thomas, C.D. (2012) Temperature- 591  \n",
      "dependent alterations in host use drive rapid range expansion in a butterfly. Science 336, 592  \n",
      "1028-1030.  593  \n",
      "65. Pauli, H., Gottfried, M., Dullinger, S., Abdaladze, O., Akhalkatsi, M., Alonso, J.L.B. et al. 594  \n",
      "(2012). Recent plant diversity changes on Europe’s mountain summits. Science 336, 353- 595  \n",
      "355.  596  \n",
      "66. Phalnikar, K., Kunte, K. & Agashe, D. (2018). Dietary and developmental shifts in butterfy- 597  \n",
      "associated bacterial communities. Royal Society Open Science 5, 171559. 598  \n",
      "67. Phalnikar, K., Kunte, K., Agashe, D. (2019). Disrupting butterfly caterpillar microbiomes 599  \n",
      "does not impact their survival and development. Proceedings of the Royal Society of London 600  \n",
      "B 286, 20192438.  601  \n",
      "68. Phillips, B.L., Kelehear, C., Pizzatto, L., Brown, G.P. Barton, D. & Shine, R. (2010). 602  \n",
      "Parasites and pathogens lag behind their host during periods of host range advance. Ecology 603  \n",
      "91(3), 872-881. 604  \n",
      "69. Qiu, Z., Coleman, M.A., Provost, E., Campbell, A.H., Kelaher, B.P., Dalton, S.J., Thomas, 605  \n",
      "T., Steinberg, P.D. & Marzinelli, E.M. (2019). Future climate change is predicted to affect 606  \n",
      "the microbiome and condition of habitat-forming kelp. Proceedings of the Royal Society of 607  \n",
      "London B 286, 20181887.  608  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 2570. R Core Team (2020). R: A language and environment for statistical computing. R 609  \n",
      "Foundation for Statistical Computing, Vienna, Austria. URL https://www.R-project.org/. 610  \n",
      "71. Raes, J., & Bork, P. (2008). Molecular eco-systems biology: towards an understanding of 611  \n",
      "community function. Nature Reviews Microbiology 6, 693-699. 612  \n",
      "72. Ramirez, K.S. et al. (2019). Range-expansion effects on the belowground plant microbiome. 613  \n",
      "Nature Ecology & Evolution 3, 604-611. 614  \n",
      "73. Rasmann, S. et al (2014). Climate-driven change in plant–insect interactions along elevation 615  \n",
      "gradients. Functional Ecology 28, 46-54. 616  \n",
      "74. Rosenzweig, M.L. (1995). Species Diversity in Space and Time. Cambridge University 617  \n",
      "Press, Cambridge, UK.  618  \n",
      "75. Schmitt, S., Angermeier, H., Schiller, R., Lindquist, N. & Hentschel, U. (2008). Molecular 619  \n",
      "microbial diversity survey of sponge reproductive stages and mechanistic insights into 620  \n",
      "vertical transmission of microbial symbionts. Applied Environmental Microbiology 74, 621  \n",
      "7694-7708.  622  \n",
      "76. Schmitt, S. et al. (2011). Assessing the complex sponge microbiota: core, variable and 623  \n",
      "species-specific bacterial communities in marine sponges. ISME Journal 6, 564-576.  624  \n",
      "77. Shively, R., Barboza, P., Doak, P. & Jung, T. S. (2018). Increased diet breadth of little 625  \n",
      "brown bats ( Myotis lucifugus ) at their northern range limit: a multimethod approach. 626  \n",
      "Canadian Journal Zoology  96, 31-38.  627  \n",
      "78. Singer, M. C. & Parmesan, C. (2021). Colonizations cause host shifts, diversification of 628  \n",
      "preferences and expansion of butterfly diet breadth. Global Change Biology 27, 3505-3518. 629  \n",
      "79. Suehiro, W. et al. (2017). Radiocarbon analysis  reveals expanded diet breadth associates 630  \n",
      "with the invasion of a predatory ant. Scientific Reports  7, 15016.  631  \n",
      "80. Tylianakis, J.M., Didham, R.K., Bascompte, J. & Wardle, D.A. (2008). Global change and 632  \n",
      "species interactions in terrestrial ecosystems. Ecology Letters 11, 1351-1363. 633  \n",
      "81. UK Butterfly Montoring Scheme (UKMBS), www.ukbms.org 634  \n",
      "82. Van Schooten, B., Godoy-Vitorino, F., McMillan, W.O. & Papa, R. (2018). Conserved 635  \n",
      "microbiota among young Heliconius butterfly species. PeerJ 6, e5502. 636  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 2683. Wahl M. et al. (2011). Re-structuring of marine communities exposed to environmental 637  \n",
      "change: a global study on the interactive effects of species and functional richness. PLoS 638  \n",
      "ONE 6, e19514. 639  \n",
      "84. Walther, R. (2004). Plants in a warmer world. Perspectives in Plant Ecology, Evolution and 640  \n",
      "Systematics 6, 169-185.  641  \n",
      "85. Wilsey, B.J. & Polley, H.W. (2004). Realistically low species evenness does not alter 642  \n",
      "grassland species richness–productivity relationships. Ecology 85, 2693-2700. 643  \n",
      "86. Yun, J.-H. et al. (2014). Insect Gut Bacterial Diversity Determined by Environmental 644  \n",
      "Habitat, Diet, Developmental Stage, and Phylogeny of Host. Applied and Environmental 645  \n",
      "Microbiology 80(17), 5254-5264. 646  \n",
      "87. Zhou, J. et al. (2011). Phylogenetic molecular ecological network of soil microbial 647  \n",
      "communities in response to elevated CO2. MBio 2, e00122-11. 648  \n",
      "  649  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 27Supplementary Materials for 650  \n",
      " 651  \n",
      "Warming-induced range expansio n effects on the diversity 652  \n",
      "and composition of the gut microbiome:  a case study with 653  \n",
      "two butterflies 654  \n",
      " 655  \n",
      "Daniel Montoya1,2 & Maaike de Jong3 656  \n",
      " 657  \n",
      "1Basque Centre for Climate Change (BC3), Edificio Sede 1, Planta 1, Parque Científico UPV- 658  \n",
      "EHU, Barrio Sarriena, Leioa, Spain 659  \n",
      "1IKERBASQUE, Basque Foundation for Science, Bilbao, Spain 660  \n",
      "3Royal Netherlands Academy of Arts and Sciences, 1011 JV Amsterdam, the Netherlands 661  \n",
      " 662  \n",
      "Correspondence to: daniel.montoya@bc3research.org 663  \n",
      " 664  \n",
      "This file includes: 665  \n",
      "Tables S1 and S3 666  \n",
      "Figures S1 and S2 667  \n",
      " 668  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 28Table S1 . Butterfly’s core-microbiome community composition in expanded margins. The butterfly core–microbiome community is defined as the subset of 6 69 \n",
      "OTUs found in butterfly hosts that are (i) present in at least two-thirds of the sites in each part of the butterfly’s range, a nd (ii) present with a relative 6 70 \n",
      "abundance ≥0.001% across the whole dataset within each part of the range. The application of stricter criteria did not significantly chang e the original results. 6 71 \n",
      "Numbers represent the percentage of the microbiome in the expanded margins that is contained in the original range of the butte rflies.  6 72 \n",
      " 6 73 \n",
      " 6 74 \n",
      " 6 75 \n",
      " 6 76 \n",
      " 6 77 \n",
      " 6 78 \n",
      "  6 79  Presence ≥ 2/3 Abundance ≥ 0.001%  \n",
      "Abundance ≥ 0.001  Abundance ≥ 0.01  Abundance ≥ 0.1 Presence = 100%  Presence ≥ 2/3 \n",
      "Brown argus  0,67 0,75 1 0,69 0,80 \n",
      "Speckled wood  0,83 0,625 1 0,88 0,79 . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 29Table S2 . β-diversity, measured as Whittaker’s index, for different abundance categories. For 680  \n",
      "any abundance category, β-diversity is always lower for the dominant bacteria and increases as 681  \n",
      "bacteria become less abundant. 682  \n",
      " 683  \n",
      "  β-diversity (Whittaker’s index) \n",
      "Abundance category Brown argus Speckled wood \n",
      " > 0.1 0.38 0.47 \n",
      "High > 0.01 0.22 0.39 \n",
      " > 0.001 0.47 0.48 \n",
      " 0.1 – 0.001 0.59 0.47 \n",
      "Intermediate 0.01 – 0.0001 0.34 0.48 \n",
      " 0.0001 – 0.00001 0.38 0.57 \n",
      " < 0.001 0.69 0.71 \n",
      "Low < 0.0001 0.75 0.71 \n",
      " < 0.00001 0.88 0.88 \n",
      " 684  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 30Table S3 . Taxa (OUT) identification for each sampling site. BA = Brown argus; SW = Speckled wood.  6 85 \n",
      " 6 86 \n",
      "6 87 \n",
      ". CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      "Figure S1 . Rarefaction curves described for (A) Observed OTUs and (B) Chao1 metric among 688  \n",
      "all sample sites. (C) Correlation between observed OTUs and rarefied OTUs. (D) Relative 689  \n",
      "abundance of OTUs for each sampling site (see table S3 for taxa ID). 690  \n",
      " 691  \n",
      " 692  \n",
      " 693  \n",
      "  694  \n",
      ". CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      "Figure S2 . Rank abundance distributions of butterflies’ microbial communities per sampling 695  \n",
      "site. The observed patterns are generally consistent to those documented when pooling all sites 696  \n",
      "together (main text). 697  \n",
      " 698  \n",
      " 699  \n",
      " 700  \n",
      " 701  \n",
      " 702  \n",
      " 703  \n",
      ". CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " 1 1 \n",
      "\n",
      "Warming-induced range expansio n effects on the diversity  3 \n",
      "and composition of the gut microbiome: 4 \n",
      "a case study with two butterflies 5 \n",
      "\n",
      "\n",
      "\n",
      "Daniel Montoya1,2 & Maaike de Jong3,4 9 \n",
      "\n",
      "\n",
      "1Basque Centre for Climate Change (BC3), Edificio Sede 1, Planta 1, Parque Científico UPV- 12  \n",
      "EHU, Barrio Sarriena, Leioa, Spain 13  \n",
      "2IKERBASQUE, Basque Foundation for Science, Bilbao, Spain 14  \n",
      "3Royal Netherlands Academy of Arts and Sciences, 1011 JV Amsterdam, the Netherlands 15  \n",
      "4Netherlands eScience Centre 16  \n",
      "\n",
      "\n",
      "Author for correspondence: 19  \n",
      "Daniel Montoya, ( daniel.montoya@bc3research.org ) 20  \n",
      "Tel: (+34)  944 01 46 90  21  \n",
      "Basque Centre for Climate Change (BC3), Edificio Sede 1, Planta 1, Parque Científico UPV- 22  \n",
      "EHU, Barrio Sarriena, Leioa, Spain 23  \n",
      "\n",
      " 25  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 2 26  \n",
      "Abstract 27  \n",
      "Species are the habitat of a diverse community of organisms, including bacteria, fungi and 28  \n",
      "viruses – the microbiome. Evidence shows that many species shift and/or expand their 29  \n",
      "geographical ranges following warming climate; yet, the response of host-associated 30  \n",
      "microbiome communities to species range shifts has received less attention, especially in 31  \n",
      "observational studies. Here, we use two British butterfly species for which detailed long-term 32  \n",
      "distributional data is available, and document for the first time a variety of effects of warming- 33  \n",
      "induced range expansion on the diversity and composition of their microbiome. Our results 34  \n",
      "show similar diversity and species-specific responses in the evenness of the gut microbial 35  \n",
      "communities in the two butterflies. However, butterflies at the expanded ranges showed (i) a 36  \n",
      "larger variability in the microbiome diversity, and (ii) a smaller core community of bacteria that 37  \n",
      "is to a large extent a subset of the bacterial core community in the original range. The 38  \n",
      "microbiome is responsible for many aspects of the host’s physiology and growth, and for 39  \n",
      "ecosystem function, so if the changes in the gut microbial communities reported here apply to 40  \n",
      "other species and taxonomic groups, the potential impact to biodiversity and functioning after 41  \n",
      "range expansion could be severe. 42  \n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Word count : Abstract: 195; Main text: 4543; References: 87; Figures: 5; Tables: 0   48  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 3Introduction 49  \n",
      "Current climate change takes place at unprecedented rates when compared to historical records 50  \n",
      "(Jones & Mann 2004), and its ecological consequences have been documented for thousands of 51  \n",
      "species worldwide (Parmesan & Yohe 2003). Among the well-documented responses to 52  \n",
      "increasing temperature are the organisms’ range shifts towards the poles and higher altitudes 53  \n",
      "(Grabherr, Gottfried & Pauli 1994; Parmesan 1996; Stange & Ayres 2001; Parmesan & Yohe 54  \n",
      "2003; Hickling et al 2006; Hoegh-Guldberg et al. 2008; Burrows et al. 2011; Chen et al 2011; 55  \n",
      "Pateman et al. 2012; Pauli et al. 2012; Rassmann et al 2014). There is ample evidence that such 56  \n",
      "range shifts can impact species diversity (Parmesan 1996, 2006), community composition 57  \n",
      "(Wahl et al 2011), body-size structure (Albouy et al 2014), phenotypic plasticity (Matesanz et 58  \n",
      "al. 2010), and species interactions (e.g. Tylianakis et al 2008; Ledger et al 2012, Memmott et al 59  \n",
      "2007; Durant et al 2007). For example, as species of different trophic levels do not necessarily 60  \n",
      "shift their ranges equally in response to climate change, established food web interactions may 61  \n",
      "become disrupted or spatially decoupled (Phillips et al 2010; Rassmann et al 2014; Gomez-Ruiz 62  \n",
      "& Lacher 2019). These asynchronous responses in space following recent climate change can 63  \n",
      "result in shifting patterns of species dominances and evenness within communities, and 64  \n",
      "potentially to the formation of non-analogue communities (Huntley 1991; Walther 2004; 65  \n",
      "Kullmann 2006; Hobbs et al 2009). Such changes can have important consequences for further 66  \n",
      "population responses to climate change and for ecosystem processes, and therefore need to be 67  \n",
      "considered in making inclusive predictions about the effects of a warming climate on 68  \n",
      "biodiversity and ecosystems (Gilman et al 2010). 69  \n",
      "\n",
      "Despite the plethora of empirical studies documenting a spatial expansion in populations 71  \n",
      "following climate change, the response of host-associated microbiome communities to species 72  \n",
      "range shifts has received less attention. Most species are a habitat for a relatively diverse 73  \n",
      "community of organisms, including bacteria, fungi and viruses. These communities are of major 74  \n",
      "influence on organismal fitness and population ecology, but until recently it was nearly 75  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 4impossible to study microbial communities at a large scale due to methodological limitations. 76  \n",
      "Metagenomic sequencing has made it possible to identify the microbiome of organic samples, 77  \n",
      "and several studies have investigated how microbial communities of host species changes to 78  \n",
      "increasing temperature. For example, recent research has found climate-induced changes in the 79  \n",
      "microbiome community of several taxonomic groups, including Sphagnum  mosses (Carrell et al 80  \n",
      "2019), kelp (Qiu et al 2019), and the gut microbiota of amphibian (Fontaine et al 2018), reptile 81  \n",
      "(Bestion et al 2017) and rodent species (Li et al 2020). Collectively, these studies provide 82  \n",
      "evidence that increasing temperatures can impact the diversity and composition of host 83  \n",
      "microbiomes, with implications to the organism’s physiology and survival, and to ecosystem 84  \n",
      "function. However, while these results come mainly from experimental mesocosms, 85  \n",
      "comparative, observational studies of host-microbiome communities in original and expanded 86  \n",
      "distributions of range-shifting species are scarce (Jones et al 2018; Ramirez et al 2019). 87  \n",
      "\n",
      "In this context, a fundamental question arises: Are there general patterns in the way the 89  \n",
      "microbiome responds to climate change-driven range shifts of the hosts, or are responses merely 90  \n",
      "idiosyncratic – i.e. each host microbiome responds differently? More specifically, what are the 91  \n",
      "effects of warming-induced range expansion on the diversity and composition of the host’s 92  \n",
      "microbiome? Several hypotheses may explain the effects of climate change on the microbial 93  \n",
      "communities of range-expanding hosts. Species undergoing range expansions may escape from 94  \n",
      "antagonistic interactions such as those from their natural enemies. For example, Engelkes et al. 95  \n",
      "(2008) showed that range-expanding plants were released from shoot and root enemy control, 96  \n",
      "whereas Menéndez et al. (2008) found similar parasitoid richness but lower parasitoid attack 97  \n",
      "rates in the expanded range of a butterfly relative to its original range. If this so-called enemy 98  \n",
      "release  hypothesis is in place, we would expect microbial communities in the expanded range to 99  \n",
      "be a subset of those in the original range. This hypothesis applies to antagonistic interactions; 100  \n",
      "yet, host-microbiome interactions can also be symbiotic. Conversely, range shifts may also lead 101  \n",
      "to the formation of new interactions in the species’ expanded range margins (e.g. Kullman 102  \n",
      "2006; Collinge & Ray 2009; Hobbs et al. 2009; Lurgi et al 2012). In other words, a shift in 103  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 5spatial distribution range may result in a parallel shift in species composition of microbial 104  \n",
      "communities. However, it remains largely unknown what is the balance between lost and new 105  \n",
      "species and interactions. Besides, most studies do not consider microbial communities, so we 106  \n",
      "know little about the potential changes in the diversity and composition of the host microbiome 107  \n",
      "following warming-induced range shifts of the host.  108  \n",
      "\n",
      "British butterflies are excellent focal species to study the ecological consequences of range 110  \n",
      "shifts. Their ecology is exceptionally well known and they are highly sensitive, and respond 111  \n",
      "rapidly to, environmental change. This is because, as other insect groups, butterflies are 112  \n",
      "ectothermic, have high reproductive rates and short generation times, and a high ability to 113  \n",
      "disperse, which lead them to respond more quickly to climate change through range shift than 114  \n",
      "longer-lived and less mobile organisms, e.g. plants. Furthermore, detailed long-term 115  \n",
      "distributional data is available for UK butterflies, which is an invaluable source for research on 116  \n",
      "the effects of climate change on species’ ranges. In a study on European butterflies, 63% had 117  \n",
      "ranges that shifted to the north by 35-240 km during the 20th century, while only 3% have 118  \n",
      "shifted to the south (Parmesan 1999). Of the species with the northern range boundary in the 119  \n",
      "UK, some have shown a dramatic range shift in the past decades. For example, the Brown 120  \n",
      "Argus ( Aricia agestis ) has shifted its UK range margin with more than 100 km in the past 25 121  \n",
      "years, and the Comma ( Polygonia c-album ) has moved northwards 220 km during the same 122  \n",
      "period (UKBMS, www.ukbms.org). Butterflies are also important model species in food web 123  \n",
      "ecology research, and particularly butterfly-parasitoid interactions are well studied in this 124  \n",
      "context (e.g. Klapwijk & Lewis 2014). To a lesser extent, butterfly-pathogen interactions are 125  \n",
      "used to study ecological disease dynamics (e.g. Bradley & Altizer 2005). Surprisingly though, 126  \n",
      "the butterflies’ gut microbiome has been less studied than that of other insect groups (Engel & 127  \n",
      "Moran 2013; Hammer et al 2014; Yun et al 2 014; Coyte et al 2015; Nair & Agashe 2016), 128  \n",
      "which contrasts with the detailed information on butterflies’ distribution patterns. 129  \n",
      " 130  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 6Here, we investigate how warming-induced range shifts impact gut microbial communities in 131  \n",
      "the expanded range of butterflies. Using two butterfly species as case models, we compare 132  \n",
      "diversity, abundance and composition of butterflies’ gut bacterial communities in original and 133  \n",
      "shifted ranges. Specifically, we address two questions: (1) How does climate change-driven 134  \n",
      "range expansion affect the gut microbial diversity of expanding butterfly species?, and (2) What 135  \n",
      "are the effects of climate-driven range expansion on the abundance patterns and the microbiome 136  \n",
      "composition in the expanded margin of butterflies relative to the original range?  137  \n",
      "\n",
      "\n",
      "Material and Methods 140  \n",
      "\n",
      "Butterfly species, site selection and field sampling 142  \n",
      "We used detailed data on the distribution and density of UK butterfly species from the UK 143  \n",
      "Butterfly Monitoring Scheme (UKBMS, www.ukbms.org), in the form of long-term (>30 years) 144  \n",
      "biweekly transect data for numerous sites across the UK. The UKBMS has documented recent 145  \n",
      "range shifts of several UK butterfly species in response to climate change, and allowed us to 146  \n",
      "carefully select the species and sites best suited to address our questions. We sampled two 147  \n",
      "butterfly species that have shown a severe shift of their northern range margin during the past 148  \n",
      "10-25 years: Brown Argus ( Aricia agestis ), and Speckled Wood ( Pararge aegeria ). We chose 149  \n",
      "sites where abundances of these butterflies have recurrently been high in previous years, thus 150  \n",
      "representing locations where the butterflies are already established, abundant, and can be easily 151  \n",
      "sampled. The same sites were chosen to sample ≥1 butterfly species if they coexisted. For each 152  \n",
      "butterfly, we selected sites that were spread out within the distribution range (both in original 153  \n",
      "and expanded margin) to avoid local effects on the microbiome. This protocol resulted in a total 154  \n",
      "of 14 sites (Figure 1): 8 sites for Brown argus (4 in original, 4 in expanded margin), and 6 sites 155  \n",
      "for Speckled wood (3 in original, 3 in expanded margin). Sampling was conducted in June- 156  \n",
      "September of 2014. Each site was visited several times over the flight period of each butterfly 157  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      "species to account for possible variation in microbial diversity due to environmental conditions 158 \n",
      "and age of the butterflies. The general flight times of the species are: Brown Argus: late July – 159 \n",
      "early September, Speckled Wood: July - early September. For each site, we collected 20 adult 160 \n",
      "individuals per butterfly species to analyse gut microbial diversity, and stored the samples at - 161 \n",
      "20˚C.  162 \n",
      "\n",
      "Figure 1 . Map of the sampling locations across the distribution range of the two butterfly 164 \n",
      "species. Sites within the expanded range are not always located north of original ranges, 165 \n",
      "despite this is the common case. For each site, 20 adult individuals per butterfly species were 166 \n",
      "collected to analyse gut microbial diversity and composition. 167 \n",
      "\n",
      "\n",
      "\n",
      "Metagenomics 171 \n",
      "After completion of the fieldwork, we extracted genomic DNA from the individual samples in 172 \n",
      "the laboratory. We focus on the gut microbiome because the likelihood of ‘contamination’ is 173 \n",
      "lower than that of external bacteria. We used Illumina High-seq to sequence the bacterial 16S 174 \n",
      "rRNA and fungal ITS ribosomal region of the pooled population samples to determine the 175 \n",
      "bacterial and fungal metagenome for each butterfly species and site. The primers were designed 176 \n",
      "with a population-specific barcode sequence so that amplification products from the different 177 \n",
      ". CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 8populations could be pooled together for a single library preparation and sequenced on a single 178  \n",
      "lane. Metagenomics sequencing was done at the University of Bristol’s Genomics Facility 179  \n",
      "(http://www.bristol.ac.uk/biology/genomics-facility/). The resulting short reads were assembled 180  \n",
      "into longer contigs, and the microbial taxonomy was determined by comparing the sequences 181  \n",
      "against online databases such as the European Ribosomal RNA database. This protocol allowed 182  \n",
      "us to identify the gut microbiome for each butterfly species – operational taxonomic unit (OTU) 183  \n",
      "– and each site.  184  \n",
      "\n",
      "Statistical and network analysis 186  \n",
      "We calculated various metrics describing the diversity, abundance and composition of microbial 187  \n",
      "communities of butterflies in original versus  expanded ranges. All analyses were performed in 188  \n",
      "R software (R version 4.0.5; R Core Team, 2020).  189  \n",
      "\n",
      "Diversity and abundance . We measured the number of OTUs (mean ± sd) and different features 191  \n",
      "of OTU abundance distributions. Changes in biological communities by anthropogenic 192  \n",
      "stressors, including climate change, not only affects the number of species, but also their 193  \n",
      "relative abundance and thereby dominance or evenness patterns in most ecosystems (Hillebrand 194  \n",
      "et al 2011). More importantly, evenness often responds more rapidly to environmental 195  \n",
      "constraints than species richness (Chapin et al 2000). Thus, the distribution and variance of 196  \n",
      "species within a host may be more important for aggregate performance of microbial 197  \n",
      "communities than the number of species. To assess how evenly OTU abundance is distributed 198  \n",
      "between original and expanded sites, we fitted Zipf models to rank-abundance distributions 199  \n",
      "(RADs). The Zipf  model  is given by the power law ar = Np1rγ, where ar is the predicted 200  \n",
      "abundance of the OTU rank r; N is the total number of individuals; p1 is the estimated 201  \n",
      "proportion of the most abundant OTU (rank 1); and γ is the estimated exponent of the power 202  \n",
      "law ( γ<0). Thus, a smaller value of | γ| indicates a more even distribution of abundances. We also 203  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 9estimated β-diversity of OTUs (Whittaker’s index) to investigate changes in OTU composition 204  \n",
      "between the butterflies’ original and expanded ranges.  205  \n",
      "\n",
      "Core communities . Ecological communities, including host-associated microbial communities, 207  \n",
      "comprise different types of microorganisms according to the intimacy and repeatability of the 208  \n",
      "association with their host, forming a continuum from core through transient to opportunistic 209  \n",
      "organisms (Schmitt et al 2011; Phalnikar et al 2019). This has prompted the development of the 210  \n",
      "‘core microbiome’ concept as a way to identify organisms that are consistently found within a 211  \n",
      "particular host species, likely performing core metabolic functions (Astudillo-Garcia et al 2017; 212  \n",
      "McDevitt-Irwin et al 2017; Lurgi et al 2019). To examine the structuring of the most prevalent 213  \n",
      "interactions between butterflies and bacteria that are in close association with them, and whether 214  \n",
      "they change following range expansion between original and shifted ranges, we constructed the 215  \n",
      "butterfly core–microbiome community. The butterfly core–microbiome community was defined 216  \n",
      "as the subset of OTUs found in butterfly hosts that are (i) present in at least two-thirds of the 217  \n",
      "sites in each part of the butterfly’s range, and (ii) present with a relative abundance ≥0.001% 218  \n",
      "across the whole dataset within each part of the range (Astudillo-García et al 2017; Lurgi et al 219  \n",
      "2019). Results were relatively robust to changing abundance criteria (see Table S1 in 220  \n",
      "Supplementary Material).  221  \n",
      "\n",
      "\n",
      "Results 224  \n",
      "\n",
      "Diversity and abundance of butterfly’s microbiome 226  \n",
      "Richness of microbial communities varies across different host species. As with most organisms 227  \n",
      "inhabiting the guts of most insect species, bacterial species comprise all or the majority of 228  \n",
      "organisms of the gut microbiome. With respect to bacterial taxonomy, our findings are 229  \n",
      "generally congruent with previous work on insect-associated bacterial communities (Engel & 230  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      "Moran 2013; Yun et al 2014; van Schooten et al 2018). The most abundant and frequent OTUs 231 \n",
      "in the studied butterflies are also common members of other insect-associated microbiomes: 232 \n",
      "Proteobacteria and Firmicutes  were the dominant phyla and represented a 50% and 21% of the 233 \n",
      "total sequences, respectively, with abundant phyla of Bacteroidetes , Actinobacteria , and 234 \n",
      "Tenericutes  in some butterfly samples (Figure S1 and Table S3 in Supplementary Material). 235 \n",
      "There was no evidence to support differences in the mean number of OTUs for butterflies at the 236 \n",
      "expanded range ( F = 1.472, p = 0.248), suggesting that range expansion did not result in a net 237 \n",
      "gain or loss of OTUs. Yet, a larger variability in the number of OTUs was observed in expanded 238 \n",
      "versus original ranges (Brown argus: {31, 55} vs {39, 53}; Speckled wood: {31, 46} vs {39, 239 \n",
      "45}) (Figure 2). 240 \n",
      "\n",
      "Figure 2 . Diversity of the butterflies’ gut microbiome in original and expanded ranges. Plots 242 \n",
      "show the mean and standard deviation of the number of bacterial OTUs for per butterfly 243 \n",
      "species. There was no evidence supporting differences in mean OTU richness (F = 1.472, p = 244 \n",
      "0.248), despite a larger variability in expanded versus original range is clear. 245 \n",
      "\n",
      "\n",
      "\n",
      "\n",
      "RAD analysis shows similar and more even distribution of OTU abundances in the butterflies’ 250 \n",
      "original range for brown argus and speckled wood, respectively ( Zipf fits, mean | γ| across sites: 251 \n",
      "Brown argus = {0.605 (original), 0.6 (expanded)}, Speckled wood = {0.61 (original), 0.67 252 \n",
      ". CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      "(expanded)}). That is, whereas no difference in evenness are observed for brown argus, certain 253 \n",
      "bacteria are more dominant in the expanded margins of the speckled wood populations (Figure 254 \n",
      "3). When divided into abundance categories we find that, whereas the abundance of dominant 255 \n",
      "bacteria is similar in both original and expanded ranges, differences in the bacterial abundance 256 \n",
      "distribution mostly arise at intermediate and low levels of abundance (Figure 4). This pattern is 257 \n",
      "consistent either aggregating all sites per butterfly or considering individual sites separately 258 \n",
      "(Figure S2 in Supplementary Material). This is further supported by the β-diversity analysis of 259 \n",
      "the three abundance categories (>0.01, 0.01 – 0.001, <0.001): despite no major turnover is 260 \n",
      "observed within the most abundant bacteria, intermediate and rare bacteria show a high turnover 261 \n",
      "(Whittaker’s index for most, intermediate and least abundant OTUs: Brown argus = {0.22, 0.34, 262 \n",
      "0.75}, Speckled wood = {0.39, 0.48, 0.71}). Although these numbers can slightly vary when 263 \n",
      "using different thresholds to establish abundance categories (Table S2 in Supplementary 264 \n",
      "Material), β-diversity is always lower for the dominant bacteria and increases as bacteria 265 \n",
      "become less abundant. 266 \n",
      "\n",
      "Figure 3 . Rank-abundance distribution of bacterial OTUs in original and expanded ranges.  (A) 268 \n",
      "Brown argus; (B) Speckled wood. Abundance values corresponding to all sites within each 269 \n",
      "range are contained within the shadows, whose limits are determined by the minimum and 270 \n",
      "maximum values across the range of bacteria abundances within each range (original and 271 \n",
      "expanded) 272 \n",
      "\n",
      "\n",
      ". CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      "\n",
      "\n",
      "Figure 4 . Rank-abundance distribution of bacterial OTUs, divided into abundance categories 277 \n",
      "(black – high, blue – intermediate, red – low), in original and expanded ranges. Upper row: 278 \n",
      "Brown argus; Bottom row: Speckled wood. Abundance categories (and their colors) are 279 \n",
      "established for bacteria within the original range and maintained in the expanded range 280 \n",
      "graphs, allowing to visualize bacterial turnover between original and expanded ranges. Most of 281 \n",
      "the differences in the abundance distribution of bacteria arise at intermediate and low levels of 282 \n",
      "abundance, whereas dominant bacteria are persistent across original and expanded ranges. 283 \n",
      "Numbers indicate β-diversity (Whittaker’s index) for the different abundance categories (higher 284 \n",
      "β indicates higher turnover). 285 \n",
      "\n",
      "\n",
      "\n",
      "Core communities  289 \n",
      "Figure 5 shows the core communities of each butterfly species with original and expanded range 290 \n",
      "sites. Despite the application of such strict constraints on the definition of the core microbiome 291 \n",
      ". CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      "(see Material and Methods), a relatively large butterfly core–microbiome community remained. 292 \n",
      "Butterflies in the original sites have a larger core community of gut bacteria relative to that of 293 \n",
      "the expanded regions. Besides, the core community at the expanded margin is, to a large extent, 294 \n",
      "contained within the original range for the two butterflies, thus suggesting that butterflies left 295 \n",
      "some bacteria behind during climate-driven range expansion. The butterfly-microbiome also 296 \n",
      "acquires new bacteria at the expanded margin, this resulting in a similar mean number of OTUs 297 \n",
      "(Figure 2). These results were robust to changing abundance criteria (see Table S1 in 298 \n",
      "Supplementary Material): for both butterflies, at least two-thirds of the core community of 299 \n",
      "bacteria at the expanded range are a subset of the core community within the original range of 300 \n",
      "distribution. 301 \n",
      "\n",
      "Figure 5 . Butterfly-bacteria core communities across distributional ranges. For each butterfly, 303 \n",
      "original and expanded range sites are included. The number of bacterial OTUs in original and 304 \n",
      "expanded range sites are indicated, as well as the number of shared bacterial OTUs across the 305 \n",
      "entire distribution range of each butterfly species. 306 \n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Discussion 312 \n",
      ". CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 14To our knowledge, this is the first observational study in real-world communities that 313  \n",
      "investigates changes in the diversity and composition of the host-microbiome in species that 314  \n",
      "have experienced warming-induced range expansion. Using two butterfly species as case 315  \n",
      "studies, our results reveal changes in butterfly microbial communities in the expanded range of 316  \n",
      "the butterflies’ distribution. Despite no significant changes in mean OTU numbers and species- 317  \n",
      "specific responses in evenness, two general patterns were observed at the shifted ranges of the 318  \n",
      "butterflies that are consistent across the two species: (i) a larger variability in the diversity of gut 319  \n",
      "bacterial communities, and (ii) a smaller core community of bacteria that is to a large extent a 320  \n",
      "subset of the bacterial core community in the original range.  321  \n",
      "\n",
      "Despite our results find support for negative effects of range shifts on the average bacterial 323  \n",
      "diversity of the gut microbiome (e.g. core communities are smaller at expanded ranges), they 324  \n",
      "are compensated with the acquisition of new bacteria. The balance between lost and gained 325  \n",
      "bacteria is neutral, thus resulting in microbiomes that are not significantly different in terms of 326  \n",
      "mean OTUs despite a larger variability in their numbers. Our analysis also reveals that the core 327  \n",
      "bacterial microbiome of the butterflies becomes less diverse after range expansion. Moreover, 328  \n",
      "the core microbiome at the expanded margin is, to a large extent, contained within the original 329  \n",
      "range for both butterflies. Such reduction in the diversity of the core bacterial communities in 330  \n",
      "parallel to similar mean, but higher diversity levels, may be driven by several, non-exclusive 331  \n",
      "factors. First, butterflies acquire their microbiome from the environment via food sources 332  \n",
      "(horizontal transmission), and thus smaller core communities at the expanded ranges of the 333  \n",
      "butterflies’ distribution would result from the fact that only a fraction of the bacteria found in 334  \n",
      "the butterflies’ core gut microbiome is common in both parts of the distribution range. 335  \n",
      "However, range-shifting populations tend to have broader diets, and this may counteract the 336  \n",
      "effects of potentially lower diversity of food sources that provide core bacteria (Braschler & 337  \n",
      "Hill 2007; Pateman et al 2012; Lancaster 2016; Eloy de Amorim et al 2017; Suehiro et al 2017; 338  \n",
      "Shively et al 2018; Singer & Parmesan 2020); indeed, Lancaster (2020) analyzed 4410 butterfly 339  \n",
      "species across the globe and found that geographic range expansions are linked to butterflies’ 340  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 15diet expansion. Collectively, our results suggest that smaller core microbiomes in the expanded 341  \n",
      "range of the butterflies’ distribution are compensated by broader diet breadths, this leading to 342  \n",
      "bacterial communities that are highly variable, yet similar in their mean OTU diversity. 343  \n",
      "\n",
      "An alternative explanation to the larger variability in bacterial diversity is that butterflies at the 345  \n",
      "expanded range are going through a transient state. On one hand, the gut microbiome is 346  \n",
      "horizontally transmitted and depends on both the availability of food resources and the 347  \n",
      "butterflies’ diet breadth. Diet breadth tends to increase with latitude and decline with population 348  \n",
      "age in butterflies (Singer & Parmesan 2020). Thus, the high variability of bacterial diversity 349  \n",
      "observed in the expanded ranges could reflect a transient state where the studied butterflies’ 350  \n",
      "populations are still young and not yet stabilised. Such variability could also be indicative of the 351  \n",
      "strength of host-bacterial interactions; in the case of symbiont interactions, low variability 352  \n",
      "would indicate more obligate interactions between butterflies and bacteria (Phalnikar et al 353  \n",
      "2019). On the other hand, it could also be argued that the studied butterflies have already 354  \n",
      "reached an equilibrium state in their expanded margins, as (i) butterflies exhibit a strong 355  \n",
      "response to warming given their ectothermic nature, short generation times, and high dispersal 356  \n",
      "ability, and (ii) we chose sites where the butterflies were already present in the last 10-25 years 357  \n",
      "and with high abundances in previous years before sampling. However, the studied butterflies 358  \n",
      "have 1-2 generations per year, so it is debatable whether their populations are at equilibrium. 359  \n",
      "Further, newer populations in the expanded margins are usually smaller and/or more isolated, 360  \n",
      "and this may reduce microbial exchange via the environment. Studies that include temporal 361  \n",
      "resolution will be needed to determine whether the observed reduction in microbiome diversity 362  \n",
      "reflects a transient state. 363  \n",
      "\n",
      "Our results show that bacteria richness and evenness can respond independently to warming- 365  \n",
      "induced range expansion, as shown by changes in the bacterial abundance distributions of one 366  \n",
      "of the butterflies’ microbiome. Specifically, whereas no difference is reported for brown argus, 367  \n",
      "bacterial abundances in the shifted ranges of speckled wood populations are less evenly 368  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 16distributed compared to the original range. This generally agrees with recent findings 369  \n",
      "documenting that species richness and abundances can respond simultaneously to global change 370  \n",
      "factors in various and independent ways (e.g. Chapin et al 2000; Hillebrand et al 2011; 371  \n",
      "McWilliams et al 2019). Our results further show that the identity of the most dominant bacteria 372  \n",
      "in the microbiome persists in both parts of the butterflies’ range, and that the effects of range 373  \n",
      "expansion on the butterflies’ microbiome has a higher impact in bacteria with intermediate and 374  \n",
      "low levels of abundance. These are the less stable bacteria that experience the highest turnover 375  \n",
      "between original and shifted ranges. Differences in the abundance distributions and evenness 376  \n",
      "patterns of bacteria could be explained by the fact  that, in butterflies, the impact of gut microbes 377  \n",
      "on their insect hosts range along a continuum from strong to weak dependence, to no 378  \n",
      "association (Phalnikar et al 2019). Moreover, even when acquired independently each 379  \n",
      "generation, insect gut communities are not expected to be random assemblages of bacteria from 380  \n",
      "the food or local environment (most studies show differences between microbes in the guts vs 381  \n",
      "those ingested with food). Rather, gut communities seem to be dominated by widely distributed 382  \n",
      "bacteria that appear to colonize hosts opportunistically (Engel & Moran 2013). Thus, whereas 383  \n",
      "rare bacteria might be more responsive to warming and/or less widely distributed, abundant gut 384  \n",
      "bacteria become abundant because they establish intimate associations with their host butterflies 385  \n",
      "(e.g. symbiosis) and/or are present in high abundances across the distribution range of 386  \n",
      "butterflies (opportunistic).  387  \n",
      "\n",
      "How do the observed changes in the microbiome diversity and composition affect butterflies’ 389  \n",
      "survival and ecosystem function? Although this is beyond the scope of this work, the results 390  \n",
      "here reported have several conservation implications. First, extinction risk decreases with 391  \n",
      "evenness, as shown by e.g. experiments where plant extinctions were higher in plots sown with 392  \n",
      "low compared to high evenness (Wilsey & Polley 2004). In the case of speckled wood 393  \n",
      "populations, this suggests that butterflies’ microbiome at the shifted ranges may lose more 394  \n",
      "microbial diversity, especially that of rare bacteria. Second, experimental studies across 395  \n",
      "different taxonomic groups show that alterations in diversity and composition of host 396  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 17microbiomes in response to warming can strongly impact the host’s physiology and survival 397  \n",
      "(Bestion et al 2017; Fontaine et al 2018; Carrel et al 2019; Qiu et al 2019; Li et al 2020). 398  \n",
      "Besides, changes in evenness are as important as changes in species richness and can have 399  \n",
      "important consequences for ecosystem function (Hillebrand et al 2011). Third, our analyses did 400  \n",
      "not discriminate between antagonistic and symbiotic bacteria, so relative changes in the 401  \n",
      "proportion of these interactions might be taking place. In contrast to the benefits associated with 402  \n",
      "the release from antagonistic interactions (e.g. predators, pathogens) on range-expanding 403  \n",
      "species (Callaway et al 2004; Mueller & Hellmann 2008; Mitchell et al 2006; Lavergne et al 404  \n",
      "2010; Hellmann et al 2012), a higher symbiont gut bacterial diversity is often beneficial to hosts 405  \n",
      "(Bolnick et al 2014). Thus, climate-driven changes in bacterial composition could be 406  \n",
      "detrimental for host butterflies, particularly if it entails a loss of essential functions (Mandrioli 407  \n",
      "2012; Le Chatelier et al 2013; Kikuchi et al 2016). 408  \n",
      "\n",
      "Our study has several limitations that future analyses should overcome. For example, we used 410  \n",
      "two butterfly species as case studies. Although these butterflies were selected because of the 411  \n",
      "availability of detailed information on their range expansion, adding more species and sites 412  \n",
      "would provide a more complete picture of the effects of warming-induced range expansion and 413  \n",
      "whether the results here reported can be generalized to other butterflies and different taxonomic 414  \n",
      "groups. Further, despite each site was visited several times over the butterflies’ flight period to 415  \n",
      "account for possible variation in microbial diversity, we acknowledge that this procedure 416  \n",
      "captures within-year variation but does not account for the long-term patterns of range 417  \n",
      "expansion. Studies should incorporate a dynamical perspective with time-series to overcome 418  \n",
      "this limitation and reveal whether range-shifting butterflies are going through a transient state or 419  \n",
      "have already reached a dynamical equilibrium within their expanded ranges. Despite these 420  \n",
      "limitations, to our knowledge this is the first comparative, observational study of host- 421  \n",
      "microbiome communities in original and expanded distributions of range-shifting species, and a 422  \n",
      "useful first step to contribute to furthering our understanding of the effects of warming on the 423  \n",
      "diversity and composition and of host bacterial communities after range expansion. 424  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 18 425  \n",
      "There is ample evidence across geographical areas and taxonomic groups on species’ range 426  \n",
      "shifts towards the poles and higher altitudes as a response to warming climate. Here, we have 427  \n",
      "used two butterfly species as case studies and documented for the first time a variety of effects 428  \n",
      "of warming-induced range expansion on the diversity and composition of their gut microbial 429  \n",
      "communities. These range from non-significant changes (bacterial richness), to idiosyncratic or 430  \n",
      "butterfly-dependent changes (evenness patterns), and changes that are consistent across the two 431  \n",
      "butterflies (variability in bacterial richness, core microbiome composition). The microbiome is 432  \n",
      "responsible for many aspects of the host’s physiology and growth, and for ecosystem function, 433  \n",
      "so if the changes in the gut microbial communities reported here apply to other species and 434  \n",
      "taxonomic groups, the potential impact to biodiversity and functioning after range expansion 435  \n",
      "could be severe. Future studies should aim at investigating the degree to which the patterns 436  \n",
      "observed in this study are consistent and the consequences of its alteration to species and 437  \n",
      "ecosystems. 438  \n",
      "\n",
      "\n",
      "Acknowledgements 441  \n",
      "We thank Emmeline Topp, Joanne Morten, Kees Wanders, Alicia Hodson and Jon Bridle for 442  \n",
      "their assistance in conducting field and lab work, and Tom Batstone for the bioinformatic 443  \n",
      "analyses. This research was financially supported by a British Ecological Society Large 444  \n",
      "Research Grant.  445  \n",
      "  446  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 19REFERENCES 447  \n",
      "\n",
      "1. Albouy, C. et al. (2013) Projected climate change and the changing biogeography of coastal 449  \n",
      "Mediterranean fishes. Journal of Biogeography 40, 534-547. 450  \n",
      "2. Astudillo-Garcia, C. et al. (2017). Evaluating the core microbiota in complex communities: a  451  \n",
      "systematic investigation. Environmental Microbiology 19, 1450-1462. 452  \n",
      "3. Bestion, E., Jacob, S., Zinger, L., Di Gesu, L., Richard, M., White, J. & Cote, J. (2017). 453  \n",
      "Climate warming reduces gut microbiota diversity in a vertebrate ectotherm. Nature Ecology 454  \n",
      "& Evolution 1(6), 0161. 455  \n",
      "4. Blüthgen, N. et al. (2006). Measuring specialization in species interaction networks. BMC 456  \n",
      "Ecology 6, 9. 457  \n",
      "5. Bolnick, D.I. et al. (2014). Individuals' diet diversity influences gut microbial diversity in 458  \n",
      "two freshwater fish (threespine stickleback and Eurasian perch). Ecology Letters 17, 979- 459  \n",
      "987. 460  \n",
      "6. Borrvall, C. et al. (2000). Biodiversity lessens the risk of cascading extinction in model food 461  \n",
      "webs. Ecology Letters 32, 131-136. 462  \n",
      "7. Bradley, C. A. & Altizer, S. (2005). Parasites hinder monarch butterfly flight: implications 463  \n",
      "for disease spread in migratory hosts. Ecology Letters 8, 290-300.  464  \n",
      "8. Braschler, B. & Hill, J. K. (2007). Role of larval host plants in the climate-driven range 465  \n",
      "expansion of the butter y Polygonia c-album . Journal of Animal Ecology  76, 415-423.  466  \n",
      "9. Burrows, M.T., Schoeman, D.S., Buckley, L.B., Moore, P., Poloczanska, E.S., Brander, 467  \n",
      "K.M. et al. (2011). The pace of shifting climate in marine and terrestrial ecosystems. Science 468  \n",
      "334, 652-655. 469  \n",
      "10. Callaway, R.M. et al. (2004). Soil biota and exotic plant invasion. Nature 427, 731-733.  470  \n",
      "11. Carrell, A.A. et al (2019). Experimental warming alters the community composition, 471  \n",
      "diversity, and N2 fixation activity of peat moss (Sphagnum fallax) microbiomes. Global 472  \n",
      "Change Biology 25, 2993-3004. 473  \n",
      "12. Chapin III, F. S. et al. (2000). Consequences of changing biodiversity. Nature 405, 234-242. 474  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 2013. Chen, I. C. et al. (2011). Rapid range shifts of species associated with high levels of climate 475  \n",
      "warming. Science 333, 1024-1026. 476  \n",
      "14. Collinge S.K. & Ray C. (2009). Transient patterns in the assembly of vernal pool plant 477  \n",
      "communities. Ecology 90, 3313-23. 478  \n",
      "15. Coyte, K.Z., Schluter, J. & Foster, K.R. (2015). The ecology of the microbiome: Networks, 479  \n",
      "competition, and stability. Science 350, 663-666. 480  \n",
      "16. Dunne, J.A. et al. (2002). Network structure and biodiversity loss in food webs: robustness 481  \n",
      "increases with connectance. Ecology Letters 5, 558-567 482  \n",
      "17. Durant, J.M., Hjermann, D.Ø., Ottersen, G. & Stenseth, N.C. (2007) Climate and the match 483  \n",
      "or mismatch between predator requirements and resource availability. Climate Research 33, 484  \n",
      "271-283.  485  \n",
      "18. Eloy de Amorim, M. et al. (2017). Lizards on newly created islands independently and 486  \n",
      "rapidly adapt in morphology and diet. Proceedings of the National Academy of Sciences 487  \n",
      "USA  114, 8812-8816.  488  \n",
      "19. Engel, P. & Moran, N.A. (2013). The gut microbiota of insects – diversity in structure and 489  \n",
      "function. FEMS Microbiology Review 37, 699-735. 490  \n",
      "20. Engelkes, T. et al. (2008). Successful range-expanding plants experience less above-ground 491  \n",
      "and below-ground enemy impact. Nature 456, 946-948. 492  \n",
      "21. Fontaine, S.S., Novarro, A.J. & Kohl, K.D. (2018). Environmental temperature alters the 493  \n",
      "digestive performance and gut microbiota of a terrestrial amphibian. Journal of Experimental 494  \n",
      "Biology 221, jeb187559. 495  \n",
      "22. Gilman, S. E. et al. (2010). A framework for community interactions under climate change. 496  \n",
      "Trends in Ecology & Evolution 25, 325-331. 497  \n",
      "23. Gomez-Ruiz, E.P. & Lacher Jr, T.E. (2019). Climate change, range shifts, and the disruption 498  \n",
      "of a pollinator-plant complex. Scientific Reports 9, 1-10. 499  \n",
      "24. Grabherr, G., Gottfried, M. & Pauli, H. (1994). Climate effects on mountain plants. Nature 500  \n",
      "369, 448.  501  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 2125. Gröbler, B.C. & O.T. Lewis. (2008). Response of native parasitoids to a range-expanding 502  \n",
      "host. Ecological Entomology 33, 453-463.  503  \n",
      "26. Hackett, T., Sauve, A., Davies, N., Montoya, D., Tylianakis, J., Memmott, J. (2019). 504  \n",
      "Reshaping our understanding of species’ roles in landscape-scale networks. Ecology Letters 505  \n",
      "22, 1367-1377. 506  \n",
      "27. Hammer, T.J., McMillan, W.O. & Fierer, N. (2014). Metamorphosis of a Butterfly- 507  \n",
      "Associated Bacterial Community. PLoS ONE 9(1), e86995. 508  \n",
      "28. Hellmann, J.J., Prior, K.M. & Pelini, S.L. (2012). The influence of species interactions on 509  \n",
      "geographic range change under climate change. Annals of the New York Academy of 510  \n",
      "Sciences 1249, 18-28. 511  \n",
      "29. Hickling R., Roy D.B., Hill J.K., Fox R., Thomas C.D. (2006). The distributions of a wide 512  \n",
      "range of taxonomic groups are expanding polewards. Global Change Biology 12, 450-455.  513  \n",
      "30. Hillebrand, H., Bennett, D.M. & Cadotte, M.A. (2008). Consequences of dominance: a 514  \n",
      "review of evenness effects on local and regional ecosystem processes. Ecology 89, 1510- 515  \n",
      "1520. 516  \n",
      "31. Hobbs, R. J., Higgs, E. & Harris, J. A. (2009). Novel ecosystems: implications for 517  \n",
      "conservation and restoration. Trends in Ecology & Evolution 24, 599-605.  518  \n",
      "32. Hoegh-Guldberg, O., Hughes, L., McIntyre, S., Lindenmayer, D.B., Parmesan, C., 519  \n",
      "Possingham, H.P. et al. (2008). Assisted colonization and rapid climate change. Science 321, 520  \n",
      "345-346. 521  \n",
      "33. Huntley, B. (1991). How plants respond to climate change— migration rates, individualism 522  \n",
      "and the consequences for plant communities. Annals of Botany 67, 15-22.  523  \n",
      "34. Jones, P. D. & Mann, M. E. (2004). Climate over past millennia. Reviews of Geophysics 42, 524  \n",
      "2. 525  \n",
      "35. Jones, J., DiBattista, J.D., Stat, M., Bunce, M., Boyce, M.C., Fairclough, D.V., Travers, M.J. 526  \n",
      "& Huggett, M.J. (2018). The Microbiome of the Gastrointestinal Tract of a Range-Shifting 527  \n",
      "Marine Herbivorous Fish. Frontiers in Microbiology 9, 1-13. 528  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 2236. Kikuchi, Y. et al. (2016). Collapse of Insect Gut Symbiosis under Simulated Climate 529  \n",
      "Change. mBio 7, e01578-16 530  \n",
      "37. Klapwijk, M.J. & Lewis, O. (2014). Spatial ecology of host-parasitoid interactions: A 531  \n",
      "threatened butterfly and its specialised parasitoid. Journal of Insect Conservation 18, 437- 532  \n",
      "445. 533  \n",
      "38. Kullmann, L. (2006) Long-term geobotanical observations of climate change impacts in the 534  \n",
      "Scandes of West-Central Sweden. Nordic Journal of Botany 24, 445-467.  535  \n",
      "39. Lancaster, L. T. (2016). Widespread range expansions shape latitudinal variation in insect 536  \n",
      "thermal limits. Nature Climate Change  6, 618-621.  537  \n",
      "40. Lancaster, L.T. (2020). Host use diversification during range shifts shapes global variation in 538  \n",
      "Lepidopteran dietary breadth. Nature Ecology & Evolution 4, 963-969. 539  \n",
      "41. Lavergne, S., Mouquet, N., Thuiller, W. & Ronce, O. (2010). Biodiversity and Climate 540  \n",
      "Change: Integrating Evolutionary and Ecological Responses of Species and Communities. 541  \n",
      "Annual Review in Ecology, Evolution & Systematics 41, 321-350. 542  \n",
      "42. Le Chatelier, E. et al. (2013). Richness of human gut microbiome correlates with metabolic 543  \n",
      "markers. Nature 500, 541-546.  544  \n",
      "43. Ledger, M. A. et al. (2013). Drought alters the structure and functioning of complex food 545  \n",
      "webs. Nature Climate Change 3, 223-227. 546  \n",
      "44. Li, G., Yin, B., Li, J., Wang, J., Wei, W., Bolnick, D.I., Wan, X., Zhu, B. & Zhang, Z. 547  \n",
      "(2020). Host-microbiota interaction helps to explain the bottom-up effects of climate change 548  \n",
      "on a small rodent species. The ISME Journal 4,  1795-1808. 549  \n",
      "45. Lurgi, M., López, B.C. & Montoya, J.M. (2012). Novel communities from climate change. 550  \n",
      "Philosophical Transactions of the Royal Society of London B 367, 2913-2922. 551  \n",
      "46. Lurgi, M., Thomas, T., Wemheuer, B., Webster, N.S. & Montoya, J.M. (2019). Modularity 552  \n",
      "and predicted functions of the global sponge-microbiome network. Nature Communications 553  \n",
      "10, 992. 554  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 2347. Macfadyen, S., Gibson, R. H., Symondson, W. O. C. & Memmot, J. (2011). Landscape 555  \n",
      "structure influences modularity patterns in farm food webs: consequences for pest control. 556  \n",
      "Ecological Applications 21, 516-524. 557  \n",
      "48. Mandrioli, M. (2012). Someone like it hot? Effects of global warming on insect immunity 558  \n",
      "and microbiota. Invertebrate Survival Journal 9, 58-63. 559  \n",
      "49. Matesanz, S., Gianoli, E. & Valladares F. (2010). Global change and the evolution of 560  \n",
      "phenotypic plasticity in plants. In: Schlichting CD, Mousseau TA (eds) Year in evolutionary 561  \n",
      "biology. Wiley-Blackwell, Malden, pp 35–55 562  \n",
      "50. McDevitt-Irwin, J. M., Baum, J. K., Garren, M. & Thurber, R. L. V. (2017). Responses of 563  \n",
      "coral-associated bacterial communities to local and global stressors. Frontiers in Marine 564  \n",
      "Science 4, 1-16.  565  \n",
      "51. McWilliam, C., Lurgi, M., Montoya, J.M. Sauve, A. & Montoya, D. (2019). The stability of 566  \n",
      "multitrophic communities under habitat loss. Nature Communications 10, 2322. 567  \n",
      "52. Memmott, J. Craze, P.G., Waser, N.M. & Price, M.V. (2007). Global warming and the 568  \n",
      "disruption of plant–pollinator interactions. Ecology Letters 10, 710-717. 569  \n",
      "53. Menéndez, R. et al. (2008). Escape from natural enemies during climate-driven range 570  \n",
      "expansion: a case study. Ecological Entomology 33, 413-421. 571  \n",
      "54. Mitchell C.E., Agrawal A.A., Bever J.D., Gilbert G.S., Hufbauer R.A., et al. (2006). Biotic 572  \n",
      "interactions and plant invasions. Ecology Letters 9, 726-40. 573  \n",
      "55. Montoya, D. (2008). Habitat loss, dispersal, and the probability of extinction of tree species. 574  \n",
      "Communicative & Integrative Biology 1(2), 146-147. 575  \n",
      "56. Montoya, D. (2021). Challenges and directions toward a general theory of ecological 576  \n",
      "recovery dynamics: A metacommunity perspective. One Earth 4(8), 1083-1094. 577  \n",
      "57. Montoya, D., Haegeman, B., Gaba, S., De Mazancourt, C. & Loreau, M. (2021). Journal of 578  \n",
      "Ecology 109(8), 2991-3006. 579  \n",
      "58. Mueller, J.M. & J.J. Hellmann. (2008). An assessment of invasion risk from assisted 580  \n",
      "migration. Conservation Biology 22, 562-567. 581  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 2459. Nair, A. & Agashe, D. (2016). Host-specific spatial and temporal variation in culturable gut 582  \n",
      "bacterial communities of dragonflies. Current Science 110(8), 1513-1523. 583  \n",
      "60. Parmesan, C. (1996). Climate and species’ range. Nature 382, 765–766.  584  \n",
      "61. Parmesan, C. (1999). Poleward shifts in geographical ranges of butterfly species associated 585  \n",
      "with regional warming. Nature 399, 579-583. 586  \n",
      "62. Parmesan, C. & Yohe, G. (2003). A globally coherent fingerprint of climate change impacts 587  \n",
      "across natural systems. Nature 421, 37-42. 588  \n",
      "63. Parmesan, C. (2006). Ecological and evolutionary responses to recent climate change. 589  \n",
      "Annual Review of Ecology, Evolution & Systematics 37, 637-669. 590  \n",
      "64. Pateman, R.M., Hill, J.K., Roy, D.B., Fox, R. & Thomas, C.D. (2012) Temperature- 591  \n",
      "dependent alterations in host use drive rapid range expansion in a butterfly. Science 336, 592  \n",
      "1028-1030.  593  \n",
      "65. Pauli, H., Gottfried, M., Dullinger, S., Abdaladze, O., Akhalkatsi, M., Alonso, J.L.B. et al. 594  \n",
      "(2012). Recent plant diversity changes on Europe’s mountain summits. Science 336, 353- 595  \n",
      "355.  596  \n",
      "66. Phalnikar, K., Kunte, K. & Agashe, D. (2018). Dietary and developmental shifts in butterfy- 597  \n",
      "associated bacterial communities. Royal Society Open Science 5, 171559. 598  \n",
      "67. Phalnikar, K., Kunte, K., Agashe, D. (2019). Disrupting butterfly caterpillar microbiomes 599  \n",
      "does not impact their survival and development. Proceedings of the Royal Society of London 600  \n",
      "B 286, 20192438.  601  \n",
      "68. Phillips, B.L., Kelehear, C., Pizzatto, L., Brown, G.P. Barton, D. & Shine, R. (2010). 602  \n",
      "Parasites and pathogens lag behind their host during periods of host range advance. Ecology 603  \n",
      "91(3), 872-881. 604  \n",
      "69. Qiu, Z., Coleman, M.A., Provost, E., Campbell, A.H., Kelaher, B.P., Dalton, S.J., Thomas, 605  \n",
      "T., Steinberg, P.D. & Marzinelli, E.M. (2019). Future climate change is predicted to affect 606  \n",
      "the microbiome and condition of habitat-forming kelp. Proceedings of the Royal Society of 607  \n",
      "London B 286, 20181887.  608  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 2570. R Core Team (2020). R: A language and environment for statistical computing. R 609  \n",
      "Foundation for Statistical Computing, Vienna, Austria. URL https://www.R-project.org/. 610  \n",
      "71. Raes, J., & Bork, P. (2008). Molecular eco-systems biology: towards an understanding of 611  \n",
      "community function. Nature Reviews Microbiology 6, 693-699. 612  \n",
      "72. Ramirez, K.S. et al. (2019). Range-expansion effects on the belowground plant microbiome. 613  \n",
      "Nature Ecology & Evolution 3, 604-611. 614  \n",
      "73. Rasmann, S. et al (2014). Climate-driven change in plant–insect interactions along elevation 615  \n",
      "gradients. Functional Ecology 28, 46-54. 616  \n",
      "74. Rosenzweig, M.L. (1995). Species Diversity in Space and Time. Cambridge University 617  \n",
      "Press, Cambridge, UK.  618  \n",
      "75. Schmitt, S., Angermeier, H., Schiller, R., Lindquist, N. & Hentschel, U. (2008). Molecular 619  \n",
      "microbial diversity survey of sponge reproductive stages and mechanistic insights into 620  \n",
      "vertical transmission of microbial symbionts. Applied Environmental Microbiology 74, 621  \n",
      "7694-7708.  622  \n",
      "76. Schmitt, S. et al. (2011). Assessing the complex sponge microbiota: core, variable and 623  \n",
      "species-specific bacterial communities in marine sponges. ISME Journal 6, 564-576.  624  \n",
      "77. Shively, R., Barboza, P., Doak, P. & Jung, T. S. (2018). Increased diet breadth of little 625  \n",
      "brown bats ( Myotis lucifugus ) at their northern range limit: a multimethod approach. 626  \n",
      "Canadian Journal Zoology  96, 31-38.  627  \n",
      "78. Singer, M. C. & Parmesan, C. (2021). Colonizations cause host shifts, diversification of 628  \n",
      "preferences and expansion of butterfly diet breadth. Global Change Biology 27, 3505-3518. 629  \n",
      "79. Suehiro, W. et al. (2017). Radiocarbon analysis  reveals expanded diet breadth associates 630  \n",
      "with the invasion of a predatory ant. Scientific Reports  7, 15016.  631  \n",
      "80. Tylianakis, J.M., Didham, R.K., Bascompte, J. & Wardle, D.A. (2008). Global change and 632  \n",
      "species interactions in terrestrial ecosystems. Ecology Letters 11, 1351-1363. 633  \n",
      "81. UK Butterfly Montoring Scheme (UKMBS), www.ukbms.org 634  \n",
      "82. Van Schooten, B., Godoy-Vitorino, F., McMillan, W.O. & Papa, R. (2018). Conserved 635  \n",
      "microbiota among young Heliconius butterfly species. PeerJ 6, e5502. 636  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 2683. Wahl M. et al. (2011). Re-structuring of marine communities exposed to environmental 637  \n",
      "change: a global study on the interactive effects of species and functional richness. PLoS 638  \n",
      "ONE 6, e19514. 639  \n",
      "84. Walther, R. (2004). Plants in a warmer world. Perspectives in Plant Ecology, Evolution and 640  \n",
      "Systematics 6, 169-185.  641  \n",
      "85. Wilsey, B.J. & Polley, H.W. (2004). Realistically low species evenness does not alter 642  \n",
      "grassland species richness–productivity relationships. Ecology 85, 2693-2700. 643  \n",
      "86. Yun, J.-H. et al. (2014). Insect Gut Bacterial Diversity Determined by Environmental 644  \n",
      "Habitat, Diet, Developmental Stage, and Phylogeny of Host. Applied and Environmental 645  \n",
      "Microbiology 80(17), 5254-5264. 646  \n",
      "87. Zhou, J. et al. (2011). Phylogenetic molecular ecological network of soil microbial 647  \n",
      "communities in response to elevated CO2. MBio 2, e00122-11. 648  \n",
      "  649  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 27Supplementary Materials for 650  \n",
      "\n",
      "Warming-induced range expansio n effects on the diversity 652  \n",
      "and composition of the gut microbiome:  a case study with 653  \n",
      "two butterflies 654  \n",
      "\n",
      "Daniel Montoya1,2 & Maaike de Jong3 656  \n",
      "\n",
      "1Basque Centre for Climate Change (BC3), Edificio Sede 1, Planta 1, Parque Científico UPV- 658  \n",
      "EHU, Barrio Sarriena, Leioa, Spain 659  \n",
      "1IKERBASQUE, Basque Foundation for Science, Bilbao, Spain 660  \n",
      "3Royal Netherlands Academy of Arts and Sciences, 1011 JV Amsterdam, the Netherlands 661  \n",
      "\n",
      "Correspondence to: daniel.montoya@bc3research.org 663  \n",
      "\n",
      "This file includes: 665  \n",
      "Tables S1 and S3 666  \n",
      "Figures S1 and S2 667  \n",
      " 668  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 28Table S1 . Butterfly’s core-microbiome community composition in expanded margins. The butterfly core–microbiome community is defined as the subset of 6 69 \n",
      "OTUs found in butterfly hosts that are (i) present in at least two-thirds of the sites in each part of the butterfly’s range, a nd (ii) present with a relative 6 70 \n",
      "abundance ≥0.001% across the whole dataset within each part of the range. The application of stricter criteria did not significantly chang e the original results. 6 71 \n",
      "Numbers represent the percentage of the microbiome in the expanded margins that is contained in the original range of the butte rflies.  6 72 \n",
      " 6 73 \n",
      " 6 74 \n",
      " 6 75 \n",
      " 6 76 \n",
      " 6 77 \n",
      " 6 78 \n",
      "  6 79  Presence ≥ 2/3 Abundance ≥ 0.001%  \n",
      "Abundance ≥ 0.001  Abundance ≥ 0.01  Abundance ≥ 0.1 Presence = 100%  Presence ≥ 2/3 \n",
      "Brown argus  0,67 0,75 1 0,69 0,80 \n",
      "Speckled wood  0,83 0,625 1 0,88 0,79 . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 29Table S2 . β-diversity, measured as Whittaker’s index, for different abundance categories. For 680  \n",
      "any abundance category, β-diversity is always lower for the dominant bacteria and increases as 681  \n",
      "bacteria become less abundant. 682  \n",
      "\n",
      "  β-diversity (Whittaker’s index) \n",
      "Abundance category Brown argus Speckled wood \n",
      " > 0.1 0.38 0.47 \n",
      "High > 0.01 0.22 0.39 \n",
      " > 0.001 0.47 0.48 \n",
      " 0.1 – 0.001 0.59 0.47 \n",
      "Intermediate 0.01 – 0.0001 0.34 0.48 \n",
      " 0.0001 – 0.00001 0.38 0.57 \n",
      " < 0.001 0.69 0.71 \n",
      "Low < 0.0001 0.75 0.71 \n",
      " < 0.00001 0.88 0.88 \n",
      " 684  . CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      " 30Table S3 . Taxa (OUT) identification for each sampling site. BA = Brown argus; SW = Speckled wood.  6 85 \n",
      " 6 86 \n",
      "6 87 \n",
      ". CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      "Figure S1 . Rarefaction curves described for (A) Observed OTUs and (B) Chao1 metric among 688  \n",
      "all sample sites. (C) Correlation between observed OTUs and rarefied OTUs. (D) Relative 689  \n",
      "abundance of OTUs for each sampling site (see table S3 for taxa ID). 690  \n",
      "\n",
      "\n",
      "\n",
      "\n",
      ". CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      "Figure S2 . Rank abundance distributions of butterflies’ microbial communities per sampling 695  \n",
      "site. The observed patterns are generally consistent to those documented when pooling all sites 696  \n",
      "together (main text). 697  \n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      ". CC-BY-NC-ND 4.0 International license available under a(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made The copyright holder for this preprint this version posted January 13, 2023. ; https://doi.org/10.1101/2023.01.11.523549doi: bioRxiv preprint \n",
      "\n"
     ]
    }
   ],
   "source": [
    "import PyPDF2\n",
    "import re\n",
    "\n",
    "pdf_path = \"/Users/a1-6/Project-helsinki/project_30years-butterfly/paper/2023.01.11.523549v1.full.pdf\"\n",
    "\n",
    "def remove_line_numbers(text):\n",
    "    # 假设行号的格式为一个空格开头，后面跟纯数字，然后两个空格\n",
    "    cleaned_text = re.sub(r'^\\s*\\d+\\s*\\s$', '', text, flags=re.MULTILINE)\n",
    "    return cleaned_text\n",
    "\n",
    "def extract_text_from_pdf(pdf_path):\n",
    "    try:\n",
    "        with open(pdf_path, 'rb') as file:\n",
    "            reader = PyPDF2.PdfReader(file)\n",
    "            text = \"\"\n",
    "            num_pages = len(reader.pages)\n",
    "            for page_num in range(num_pages):\n",
    "                page = reader.pages[page_num]\n",
    "                page_text = page.extract_text()\n",
    "                if page_text:\n",
    "                    page_text = remove_line_numbers(page_text)\n",
    "                    text += page_text + \"\\n\"\n",
    "            return text\n",
    "    except Exception as e:\n",
    "        print(f\"Error reading {pdf_path}: {e}\")\n",
    "        return None\n",
    "\n",
    "# 使用示例\n",
    "text = extract_text_from_pdf(pdf_path)\n",
    "if text:\n",
    "    print(text)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Extracted Abstract:\n",
      " \n"
     ]
    }
   ],
   "source": [
    "import fitz  # PyMuPDF\n",
    "import re\n",
    "\n",
    "def extract_abstract_from_pdf(pdf_path):\n",
    "    # Open the PDF document\n",
    "    document = fitz.open(pdf_path)\n",
    "    abstract_text = \"\"\n",
    "    text = \"\"\n",
    "    \n",
    "    # Iterate through each page to find the abstract\n",
    "    for page_num in range(document.page_count):\n",
    "        page = document.load_page(page_num)\n",
    "        text += page.get_text(\"text\") + \"\\n\"\n",
    "    \n",
    "    # Use regex to find the abstract section\n",
    "    match = re.search(r\"Abstract\\s+([\\s\\S]*?)(Introduction|INTRODUCTION|Materials and Methods|METHODS)\", text, re.IGNORECASE)\n",
    "    if match:\n",
    "        abstract_text = match.group(1).strip()\n",
    "        \n",
    "        # Remove line numbers and excessive newlines\n",
    "        abstract_text = re.sub(r'\\n\\s*\\d+\\s*\\n', '\\n', abstract_text)  # Remove line numbers\n",
    "        abstract_text = re.sub(r'\\n+', ' ', abstract_text)  # Replace multiple newlines with a single space\n",
    "    \n",
    "    return abstract_text\n",
    "\n",
    "# Path to the PDF file\n",
    "pdf_path = \"/Users/a1-6/Project-helsinki/project_30years-butterfly/paper/li-et-al-2024-dual-stressors-of-infection-and-warming-can-destabilize-host-microbiomes.pdf\"\n",
    "abstract = extract_abstract_from_pdf(pdf_path)\n",
    "print(\"Extracted Abstract:\\n\", abstract)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Extracted Abstract:\n",
      " \n"
     ]
    }
   ],
   "source": [
    "import fitz  # PyMuPDF\n",
    "import re\n",
    "\n",
    "def extract_abstract_from_pdf(pdf_path):\n",
    "    # Open the PDF document\n",
    "    document = fitz.open(pdf_path)\n",
    "    text = \"\"\n",
    "    \n",
    "    # Iterate through each page to find the abstract\n",
    "    for page_num in range(document.page_count):\n",
    "        page = document.load_page(page_num)\n",
    "        text += page.get_text(\"text\") + \"\\n\"\n",
    "    \n",
    "    # Use regex to find the abstract section\n",
    "    match = re.search(r\"Abstract\\n([\\s\\S]*?)(Introduction|INTRODUCTION|1\\. Introduction|2\\. Methods|Materials and Methods|METHODS)\", text, re.DOTALL | re.IGNORECASE)\n",
    "    if match:\n",
    "        abstract_text = match.group(1).strip()\n",
    "        \n",
    "        # Remove line numbers and excessive newlines\n",
    "        abstract_text = re.sub(r'\\n\\s*\\d+\\s*', ' ', abstract_text)  # Remove line numbers\n",
    "        abstract_text = re.sub(r'\\s+', ' ', abstract_text)  # Replace multiple spaces/newlines with a single space\n",
    "        \n",
    "        # Remove any remaining artifacts at the end of the abstract\n",
    "        abstract_text = re.sub(r'(Word count:.*|CC-BY.*|doi:.*|bioRxiv preprint.*| \\d+ |^\\d+ )', '', abstract_text).strip()\n",
    "        abstract_text = re.sub(r'^\\d+\\s*', '', abstract_text)  # Remove leading numbers\n",
    "        abstract_text = re.sub(r'\\s+\\d+\\s*$', '', abstract_text)  # Remove trailing numbers\n",
    "        \n",
    "        return abstract_text\n",
    "    \n",
    "    return \"\"\n",
    "\n",
    "# Path to the PDF file\n",
    "pdf_path = \"/Users/a1-6/Project-helsinki/project_30years-butterfly/paper/li-et-al-2024-dual-stressors-of-infection-and-warming-can-destabilize-host-microbiomes.pdf\"\n",
    "abstract = extract_abstract_from_pdf(pdf_path)\n",
    "print(\"Extracted Abstract:\\n\", abstract)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Extracted Abstract:\n",
      " \n"
     ]
    }
   ],
   "source": [
    "import fitz  # PyMuPDF\n",
    "import re\n",
    "\n",
    "def extract_abstract_from_pdf(pdf_path):\n",
    "    # Open the PDF document\n",
    "    document = fitz.open(pdf_path)\n",
    "    text = \"\"\n",
    "    \n",
    "    # Iterate through each page to extract text\n",
    "    for page_num in range(document.page_count):\n",
    "        page = document.load_page(page_num)\n",
    "        text += page.get_text(\"text\") + \"\\n\"\n",
    "    \n",
    "    # Use regex to find the abstract section\n",
    "    match = re.search(r\"Abstract\\s+([\\s\\S]*?)(Introduction|INTRODUCTION|1\\. Introduction|2\\. Methods|Materials and Methods|METHODS)\", text, re.DOTALL | re.IGNORECASE)\n",
    "    if match:\n",
    "        abstract_text = match.group(1).strip()\n",
    "        \n",
    "        # Clean up the abstract text\n",
    "        abstract_text = re.sub(r'\\n\\s*\\d+\\s*', ' ', abstract_text)  # Remove line numbers\n",
    "        abstract_text = re.sub(r'\\s+', ' ', abstract_text)  # Replace multiple spaces/newlines with a single space\n",
    "        abstract_text = re.sub(r'(Word count:.*|CC-BY.*|doi:.*|bioRxiv preprint.*| \\d+ |^\\d+ )', '', abstract_text).strip()  # Remove artifacts\n",
    "        abstract_text = re.sub(r'^\\d+\\s*', '', abstract_text)  # Remove leading numbers\n",
    "        abstract_text = re.sub(r'\\s+\\d+\\s*$', '', abstract_text)  # Remove trailing numbers\n",
    "        \n",
    "        return abstract_text\n",
    "    \n",
    "    return \"\"\n",
    "\n",
    "# Path to the PDF file\n",
    "pdf_path = \"/Users/a1-6/Project-helsinki/project_30years-butterfly/paper/li-et-al-2024-dual-stressors-of-infection-and-warming-can-destabilize-host-microbiomes.pdf\"\n",
    "abstract = extract_abstract_from_pdf(pdf_path)\n",
    "print(\"Extracted Abstract:\\n\", abstract)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: flask in /Users/a1-6/anaconda3/lib/python3.11/site-packages (2.2.2)\n",
      "Requirement already satisfied: pymupdf in /Users/a1-6/anaconda3/lib/python3.11/site-packages (1.24.4)\n",
      "Requirement already satisfied: Werkzeug>=2.2.2 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from flask) (2.2.3)\n",
      "Requirement already satisfied: Jinja2>=3.0 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from flask) (3.1.2)\n",
      "Requirement already satisfied: itsdangerous>=2.0 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from flask) (2.0.1)\n",
      "Requirement already satisfied: click>=8.0 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from flask) (8.0.4)\n",
      "Requirement already satisfied: PyMuPDFb==1.24.3 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from pymupdf) (1.24.3)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from Jinja2>=3.0->flask) (2.1.1)\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install flask pymupdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "no such file: 'path/to/your/pdf_file.pdf'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/0v/m9_w9cqn4nbbtjvqh38v8ykw0000gn/T/ipykernel_94071/1604387258.py\u001b[0m in \u001b[0;36m?\u001b[0;34m()\u001b[0m\n\u001b[1;32m     24\u001b[0m \u001b[0;31m# 示例PDF文件路径\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     25\u001b[0m \u001b[0mpdf_path\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'path/to/your/pdf_file.pdf'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     26\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     27\u001b[0m \u001b[0;31m# 提取文本并生成摘要\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 28\u001b[0;31m \u001b[0mextracted_text\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mextract_text_from_first_page\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpdf_path\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     29\u001b[0m \u001b[0msummary\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msummarize_text\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mextracted_text\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     30\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     31\u001b[0m \u001b[0;31m# 输出摘要\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/var/folders/0v/m9_w9cqn4nbbtjvqh38v8ykw0000gn/T/ipykernel_94071/1604387258.py\u001b[0m in \u001b[0;36m?\u001b[0;34m(pdf_path)\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mextract_text_from_first_page\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpdf_path\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m     \u001b[0;31m# 打开PDF文件\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 8\u001b[0;31m     \u001b[0mdocument\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfitz\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpdf_path\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      9\u001b[0m     \u001b[0;31m# 提取第一页的文本\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     10\u001b[0m     \u001b[0mfirst_page\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdocument\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mload_page\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     11\u001b[0m     \u001b[0mtext\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfirst_page\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_text\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pymupdf/__init__.py\u001b[0m in \u001b[0;36m?\u001b[0;34m(self, filename, stream, filetype, rect, width, height, fontsize)\u001b[0m\n\u001b[1;32m   2833\u001b[0m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpage_count2\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mextra\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpage_count_pdf\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2834\u001b[0m                 \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2835\u001b[0m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpage_count2\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mextra\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpage_count_fz\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2836\u001b[0m         \u001b[0;32mfinally\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2837\u001b[0;31m             \u001b[0mJM_mupdf_show_errors\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mJM_mupdf_show_errors_old\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m: no such file: 'path/to/your/pdf_file.pdf'"
     ]
    }
   ],
   "source": [
    "import fitz  # PyMuPDF\n",
    "import openai\n",
    "\n",
    "# 设置OpenAI API密钥\n",
    "openai.api_key = \n",
    "def extract_text_from_first_page(pdf_path):\n",
    "    # 打开PDF文件\n",
    "    document = fitz.open(pdf_path)\n",
    "    # 提取第一页的文本\n",
    "    first_page = document.load_page(0)\n",
    "    text = first_page.get_text()\n",
    "    return text\n",
    "\n",
    "def summarize_text(text):\n",
    "    # 使用OpenAI的API生成摘要\n",
    "    response = openai.Completion.create(\n",
    "        engine=\"davinci-codex\",\n",
    "        prompt=f\"Please summarize the following text:\\n\\n{text}\",\n",
    "        max_tokens=100\n",
    "    )\n",
    "    summary = response.choices[0].text.strip()\n",
    "    return summary\n",
    "\n",
    "# 示例PDF文件路径\n",
    "pdf_path = 'path/to/your/pdf_file.pdf'\n",
    "\n",
    "# 提取文本并生成摘要\n",
    "extracted_text = extract_text_from_first_page(pdf_path)\n",
    "summary = summarize_text(extracted_text)\n",
    "\n",
    "# 输出摘要\n",
    "print(\"摘要内容：\")\n",
    "print(summary)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: pymupdf in /Users/a1-6/anaconda3/lib/python3.11/site-packages (1.24.4)\n",
      "Requirement already satisfied: PyMuPDFb==1.24.3 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from pymupdf) (1.24.3)\n",
      "Collecting pdfplumber\n",
      "  Obtaining dependency information for pdfplumber from https://files.pythonhosted.org/packages/ee/ec/315015b6c4667d889cc14348c93e31f9fd43a6f61162b85c39e4a21018d5/pdfplumber-0.11.0-py3-none-any.whl.metadata\n",
      "  Downloading pdfplumber-0.11.0-py3-none-any.whl.metadata (39 kB)\n",
      "Collecting pdfminer.six==20231228 (from pdfplumber)\n",
      "  Obtaining dependency information for pdfminer.six==20231228 from https://files.pythonhosted.org/packages/eb/9c/e46fe7502b32d7db6af6e36a9105abb93301fa1ec475b5ddcba8b35ae23a/pdfminer.six-20231228-py3-none-any.whl.metadata\n",
      "  Downloading pdfminer.six-20231228-py3-none-any.whl.metadata (4.2 kB)\n",
      "Requirement already satisfied: Pillow>=9.1 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from pdfplumber) (9.4.0)\n",
      "Collecting pypdfium2>=4.18.0 (from pdfplumber)\n",
      "  Obtaining dependency information for pypdfium2>=4.18.0 from https://files.pythonhosted.org/packages/21/8b/27d4d5409f3c76b985f4ee4afe147b606594411e15ac4dc1c3363c9a9810/pypdfium2-4.30.0-py3-none-macosx_11_0_arm64.whl.metadata\n",
      "  Downloading pypdfium2-4.30.0-py3-none-macosx_11_0_arm64.whl.metadata (48 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m48.5/48.5 kB\u001b[0m \u001b[31m4.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: charset-normalizer>=2.0.0 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from pdfminer.six==20231228->pdfplumber) (2.0.4)\n",
      "Requirement already satisfied: cryptography>=36.0.0 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from pdfminer.six==20231228->pdfplumber) (41.0.3)\n",
      "Requirement already satisfied: cffi>=1.12 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from cryptography>=36.0.0->pdfminer.six==20231228->pdfplumber) (1.15.1)\n",
      "Requirement already satisfied: pycparser in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from cffi>=1.12->cryptography>=36.0.0->pdfminer.six==20231228->pdfplumber) (2.21)\n",
      "Downloading pdfplumber-0.11.0-py3-none-any.whl (56 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m56.4/56.4 kB\u001b[0m \u001b[31m4.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading pdfminer.six-20231228-py3-none-any.whl (5.6 MB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m5.6/5.6 MB\u001b[0m \u001b[31m5.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0m\n",
      "\u001b[?25hDownloading pypdfium2-4.30.0-py3-none-macosx_11_0_arm64.whl (2.7 MB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m2.7/2.7 MB\u001b[0m \u001b[31m5.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0m\n",
      "\u001b[?25hInstalling collected packages: pypdfium2, pdfminer.six, pdfplumber\n",
      "Successfully installed pdfminer.six-20231228 pdfplumber-0.11.0 pypdfium2-4.30.0\n"
     ]
    }
   ],
   "source": [
    "# 安装所需的库\n",
    "def install_packages():\n",
    "    subprocess.check_call([\"pip\", \"install\", \"pymupdf\"])\n",
    "    subprocess.check_call([\"pip\", \"install\", \"pdfplumber\"])\n",
    "\n",
    "install_packages()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dual stressors of infection and warming\n",
      "can destabilize host microbiomes\n",
      "royalsocietypublishing.org/journal/rstb\n",
      "J. D. Li1, Y. Y. Gao2,3, E. J. Stevens1 and K. C. King1,4,5\n",
      "1DepartmentofBiology, UniversityofOxford, Oxford OX1 2JD,UK\n",
      "2ShenzhenBranch,GuangdongLaboratoryofLingnanModernAgriculture,GenomeAnalysisLaboratoryofthe\n",
      "MinistryofAgricultureand Rural Affairs, Agricultural Genomics Institute at Shenzhen, ChineseAcademyof\n",
      "Research\n",
      "AgriculturalSciences, Shenzhen,Guangdong 518120, People’s RepublicofChina\n",
      "3SchoolofEcologyandNatureConservation,BeijingForestryUniversity,35TsinghuaEastRoad,Beijing100083,\n",
      "Cite this article: Li JD, Gao YY, Stevens EJ, People’s Republicof China\n",
      "4DepartmentofZoology, UniversityofBritishColumbia, Vancouver,V6T 1Z4, Canada\n",
      "King KC. 2024 Dual stressors of infection and\n",
      "5DepartmentofMicrobiology& Immunology,UniversityofBritish Columbia, Vancouver, V6T 1Z3,Canada\n",
      "warming can destabilize host microbiomes.\n",
      "JDL, 0000-0001-5426-590X; EJS,0000-0002-5299-4401;KCK, 0000-0003-1393-9220\n",
      "Phil. Trans. R. Soc. B 379: 20230069.\n",
      "https://doi.org/10.1098/rstb.2023.0069\n",
      "Climatechangeiscausingextremeheatingeventsandintensifyinginfectious\n",
      "diseaseoutbreaks.Animals harbourmicrobialcommunities,whicharevital\n",
      "Received: 10 August 2023 fortheirsurvivalandfitnessunderstressfulconditions.Understandinghow\n",
      "microbiomestructureschangeinresponsetoinfectionandwarmingmaybe\n",
      "Accepted: 2 January 2024\n",
      "important for forecasting host performance under global change. Here, we\n",
      "evaluated alterations in the microbiomes of several wild Caenorhabditis\n",
      "One contribution of 18 to a theme issue elegans isolates spanning a range of latitudes, upon warming temperatures\n",
      "‘Sculpting the microbiome: how host factors andinfectionbytheparasiteLeucobactermusarum.Using16SrRNAsequen-\n",
      "determine and respond to microbial cing, we found that microbiome diversity decreased, and dispersion\n",
      "colonization’. increased over time, with the former being more prominent in uninfected\n",
      "adults and the latter aggravated by infection. Infection reduced dominance\n",
      "of specific microbial taxa, and increased microbiome dispersion, indicating\n",
      "Subject Areas: destabilizingeffectsonhostmicrobialcommunities.Exposinginfectedhosts\n",
      "ecology towarmingdidnothaveanadditivedestabilizingeffectontheirmicrobiomes.\n",
      "Moreover,warmingduringpre-adultdevelopmentalleviatedthedestabilizing\n",
      "effectsofinfectiononhostmicrobiomes.Theseresultsrevealedanopposing\n",
      "Keywords:\n",
      "interaction between biotic and abiotic factors on microbiome structure.\n",
      "microbiomes, parasite, C. elegans, global\n",
      "Lastly,weshowedthatincreasedmicrobiomedispersionmightbeassociated\n",
      "climate change with decreased variability in microbial species interaction strength. Overall,\n",
      "these findings improve our understanding of animal microbiome dynamics\n",
      "amidstconcurrentclimatechangeandepidemics.\n",
      "Author for correspondence:\n",
      "This article is part of the theme issue ‘Sculpting the microbiome: how\n",
      "J. D. Li\n",
      "host factorsdetermine and respond to microbialcolonization’.\n",
      "e-mail: jingdi.li@biology.ox.ac.uk\n",
      "1. Introduction\n",
      "Global climate change has led to multiple climate hazards including more\n",
      "extreme temperatures, resulting in population decline and biodiversity loss [1].\n",
      "Shifting global temperatures is also changing the geographical distribution of\n",
      "infectiousdiseases[2,3].Astemperaturesincrease,hostsandparasitesareexperi-\n",
      "encingshiftsintheirthermalenvironment,drivingvariationindiseaseoutcomes\n",
      "[4,5]. Thus, projections of species persistence in the changing world will need\n",
      "to account for threats posed by both warming and infection, as well as the\n",
      "interactionofthesedualstressors[6,7].\n",
      "Accumulatingevidenceshowsthathosthealthinthefaceofchangingtemp-\n",
      "eratures can be mediated by host microbiomes [8,9] and parasite infection\n",
      "[10,11]. Microbiomes are highly sensitive to biotic or abiotic disturbances\n",
      "[12–14].Changesinmicrobiomestructureandstability,asaresult,areincreas-\n",
      "ingly recognized as meaningful indicators of altered host health [12,13,15].\n",
      "Electronic supplementary material is available\n",
      "Studying microbiome dynamics under warming and infection scenarios\n",
      "online at https://doi.org/10.6084/m9.figshare.\n",
      "c.7075485.\n",
      "©2024TheAuthors.PublishedbytheRoyalSocietyunderthetermsoftheCreativeCommonsAttribution\n",
      "License http://creativecommons.org/licenses/by/4.0/, which permits unrestricted use, provided the original\n",
      "authorandsourcearecredited.\n",
      "4202\n",
      "yaM\n",
      "32\n",
      "no\n",
      "/gro.gnihsilbupyteicoslayor//:sptth\n",
      "morf\n",
      "dedaolnwoD\n"
     ]
    }
   ],
   "source": [
    "import pdfplumber\n",
    "import os\n",
    "filepath=\"/Users/a1-6/Project-helsinki/project_30years-butterfly/paper/li-et-al-2024-dual-stressors-of-infection-and-warming-can-destabilize-host-microbiomes.pdf\"\n",
    "pdf = pdfplumber.open(filepath)\n",
    "page = pdf.pages[0]\n",
    "print(page.extract_text())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: pymupdf in /Users/a1-6/anaconda3/lib/python3.11/site-packages (1.24.4)\n",
      "Requirement already satisfied: PyMuPDFb==1.24.3 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from pymupdf) (1.24.3)\n",
      "Requirement already satisfied: openai in /Users/a1-6/anaconda3/lib/python3.11/site-packages (1.10.0)\n",
      "Requirement already satisfied: anyio<5,>=3.5.0 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from openai) (3.5.0)\n",
      "Requirement already satisfied: distro<2,>=1.7.0 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from openai) (1.9.0)\n",
      "Requirement already satisfied: httpx<1,>=0.23.0 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from openai) (0.23.0)\n",
      "Requirement already satisfied: pydantic<3,>=1.9.0 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from openai) (1.10.8)\n",
      "Requirement already satisfied: sniffio in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from openai) (1.3.0)\n",
      "Requirement already satisfied: tqdm>4 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from openai) (4.65.0)\n",
      "Requirement already satisfied: typing-extensions<5,>=4.7 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from openai) (4.9.0)\n",
      "Requirement already satisfied: idna>=2.8 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from anyio<5,>=3.5.0->openai) (3.4)\n",
      "Requirement already satisfied: certifi in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from httpx<1,>=0.23.0->openai) (2023.11.17)\n",
      "Requirement already satisfied: rfc3986[idna2008]<2,>=1.3 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from httpx<1,>=0.23.0->openai) (1.4.0)\n",
      "Requirement already satisfied: httpcore<0.16.0,>=0.15.0 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from httpx<1,>=0.23.0->openai) (0.15.0)\n",
      "Requirement already satisfied: h11<0.13,>=0.11 in /Users/a1-6/anaconda3/lib/python3.11/site-packages (from httpcore<0.16.0,>=0.15.0->httpx<1,>=0.23.0->openai) (0.12.0)\n"
     ]
    },
    {
     "ename": "APIRemovedInV1",
     "evalue": "\n\nYou tried to access openai.ChatCompletion, but this is no longer supported in openai>=1.0.0 - see the README at https://github.com/openai/openai-python for the API.\n\nYou can run `openai migrate` to automatically upgrade your codebase to use the 1.0.0 interface. \n\nAlternatively, you can pin your installation to the old version, e.g. `pip install openai==0.28`\n\nA detailed migration guide is available here: https://github.com/openai/openai-python/discussions/742\n",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAPIRemovedInV1\u001b[0m                            Traceback (most recent call last)",
      "\u001b[1;32m/Users/a1-6/Project-helsinki/M_Ms/pdf_api/ReadPDF.ipynb Cell 13\u001b[0m line \u001b[0;36m4\n\u001b[1;32m     <a href='vscode-notebook-cell:/Users/a1-6/Project-helsinki/M_Ms/pdf_api/ReadPDF.ipynb#X14sZmlsZQ%3D%3D?line=38'>39</a>\u001b[0m \u001b[39m# 提取文本并生成摘要\u001b[39;00m\n\u001b[1;32m     <a href='vscode-notebook-cell:/Users/a1-6/Project-helsinki/M_Ms/pdf_api/ReadPDF.ipynb#X14sZmlsZQ%3D%3D?line=39'>40</a>\u001b[0m extracted_text \u001b[39m=\u001b[39m extract_text_from_first_page(pdf_path)\n\u001b[0;32m---> <a href='vscode-notebook-cell:/Users/a1-6/Project-helsinki/M_Ms/pdf_api/ReadPDF.ipynb#X14sZmlsZQ%3D%3D?line=40'>41</a>\u001b[0m summary \u001b[39m=\u001b[39m summarize_text(extracted_text)\n\u001b[1;32m     <a href='vscode-notebook-cell:/Users/a1-6/Project-helsinki/M_Ms/pdf_api/ReadPDF.ipynb#X14sZmlsZQ%3D%3D?line=42'>43</a>\u001b[0m \u001b[39m# 输出摘要\u001b[39;00m\n\u001b[1;32m     <a href='vscode-notebook-cell:/Users/a1-6/Project-helsinki/M_Ms/pdf_api/ReadPDF.ipynb#X14sZmlsZQ%3D%3D?line=43'>44</a>\u001b[0m \u001b[39mprint\u001b[39m(\u001b[39m\"\u001b[39m\u001b[39m摘要内容：\u001b[39m\u001b[39m\"\u001b[39m)\n",
      "\u001b[1;32m/Users/a1-6/Project-helsinki/M_Ms/pdf_api/ReadPDF.ipynb Cell 13\u001b[0m line \u001b[0;36m2\n\u001b[1;32m     <a href='vscode-notebook-cell:/Users/a1-6/Project-helsinki/M_Ms/pdf_api/ReadPDF.ipynb#X14sZmlsZQ%3D%3D?line=23'>24</a>\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39msummarize_text\u001b[39m(text):\n\u001b[1;32m     <a href='vscode-notebook-cell:/Users/a1-6/Project-helsinki/M_Ms/pdf_api/ReadPDF.ipynb#X14sZmlsZQ%3D%3D?line=24'>25</a>\u001b[0m     \u001b[39m# 使用OpenAI的API生成摘要\u001b[39;00m\n\u001b[0;32m---> <a href='vscode-notebook-cell:/Users/a1-6/Project-helsinki/M_Ms/pdf_api/ReadPDF.ipynb#X14sZmlsZQ%3D%3D?line=25'>26</a>\u001b[0m     response \u001b[39m=\u001b[39m openai\u001b[39m.\u001b[39mChatCompletion\u001b[39m.\u001b[39mcreate(\n\u001b[1;32m     <a href='vscode-notebook-cell:/Users/a1-6/Project-helsinki/M_Ms/pdf_api/ReadPDF.ipynb#X14sZmlsZQ%3D%3D?line=26'>27</a>\u001b[0m         model\u001b[39m=\u001b[39m\u001b[39m\"\u001b[39m\u001b[39mgpt-3.5-turbo\u001b[39m\u001b[39m\"\u001b[39m,\n\u001b[1;32m     <a href='vscode-notebook-cell:/Users/a1-6/Project-helsinki/M_Ms/pdf_api/ReadPDF.ipynb#X14sZmlsZQ%3D%3D?line=27'>28</a>\u001b[0m         messages\u001b[39m=\u001b[39m[\n\u001b[1;32m     <a href='vscode-notebook-cell:/Users/a1-6/Project-helsinki/M_Ms/pdf_api/ReadPDF.ipynb#X14sZmlsZQ%3D%3D?line=28'>29</a>\u001b[0m             {\u001b[39m\"\u001b[39m\u001b[39mrole\u001b[39m\u001b[39m\"\u001b[39m: \u001b[39m\"\u001b[39m\u001b[39msystem\u001b[39m\u001b[39m\"\u001b[39m, \u001b[39m\"\u001b[39m\u001b[39mcontent\u001b[39m\u001b[39m\"\u001b[39m: \u001b[39m\"\u001b[39m\u001b[39mYou are a helpful assistant.\u001b[39m\u001b[39m\"\u001b[39m},\n\u001b[1;32m     <a href='vscode-notebook-cell:/Users/a1-6/Project-helsinki/M_Ms/pdf_api/ReadPDF.ipynb#X14sZmlsZQ%3D%3D?line=29'>30</a>\u001b[0m             {\u001b[39m\"\u001b[39m\u001b[39mrole\u001b[39m\u001b[39m\"\u001b[39m: \u001b[39m\"\u001b[39m\u001b[39muser\u001b[39m\u001b[39m\"\u001b[39m, \u001b[39m\"\u001b[39m\u001b[39mcontent\u001b[39m\u001b[39m\"\u001b[39m: \u001b[39mf\u001b[39m\u001b[39m\"\u001b[39m\u001b[39mPlease summarize the following text:\u001b[39m\u001b[39m\\n\u001b[39;00m\u001b[39m\\n\u001b[39;00m\u001b[39m{\u001b[39;00mtext\u001b[39m}\u001b[39;00m\u001b[39m\"\u001b[39m}\n\u001b[1;32m     <a href='vscode-notebook-cell:/Users/a1-6/Project-helsinki/M_Ms/pdf_api/ReadPDF.ipynb#X14sZmlsZQ%3D%3D?line=30'>31</a>\u001b[0m         ]\n\u001b[1;32m     <a href='vscode-notebook-cell:/Users/a1-6/Project-helsinki/M_Ms/pdf_api/ReadPDF.ipynb#X14sZmlsZQ%3D%3D?line=31'>32</a>\u001b[0m     )\n\u001b[1;32m     <a href='vscode-notebook-cell:/Users/a1-6/Project-helsinki/M_Ms/pdf_api/ReadPDF.ipynb#X14sZmlsZQ%3D%3D?line=32'>33</a>\u001b[0m     summary \u001b[39m=\u001b[39m response[\u001b[39m'\u001b[39m\u001b[39mchoices\u001b[39m\u001b[39m'\u001b[39m][\u001b[39m0\u001b[39m][\u001b[39m'\u001b[39m\u001b[39mmessage\u001b[39m\u001b[39m'\u001b[39m][\u001b[39m'\u001b[39m\u001b[39mcontent\u001b[39m\u001b[39m'\u001b[39m]\u001b[39m.\u001b[39mstrip()\n\u001b[1;32m     <a href='vscode-notebook-cell:/Users/a1-6/Project-helsinki/M_Ms/pdf_api/ReadPDF.ipynb#X14sZmlsZQ%3D%3D?line=33'>34</a>\u001b[0m     \u001b[39mreturn\u001b[39;00m summary\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/openai/lib/_old_api.py:39\u001b[0m, in \u001b[0;36mAPIRemovedInV1Proxy.__call__\u001b[0;34m(self, *_args, **_kwargs)\u001b[0m\n\u001b[1;32m     38\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39m__call__\u001b[39m(\u001b[39mself\u001b[39m, \u001b[39m*\u001b[39m_args: Any, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39m_kwargs: Any) \u001b[39m-\u001b[39m\u001b[39m>\u001b[39m Any:\n\u001b[0;32m---> 39\u001b[0m     \u001b[39mraise\u001b[39;00m APIRemovedInV1(symbol\u001b[39m=\u001b[39m\u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_symbol)\n",
      "\u001b[0;31mAPIRemovedInV1\u001b[0m: \n\nYou tried to access openai.ChatCompletion, but this is no longer supported in openai>=1.0.0 - see the README at https://github.com/openai/openai-python for the API.\n\nYou can run `openai migrate` to automatically upgrade your codebase to use the 1.0.0 interface. \n\nAlternatively, you can pin your installation to the old version, e.g. `pip install openai==0.28`\n\nA detailed migration guide is available here: https://github.com/openai/openai-python/discussions/742\n"
     ]
    }
   ],
   "source": [
    "from PyPDF2 import PdfReader\n",
    "filepath=\"/Users/a1-6/Project-helsinki/project_30years-butterfly/paper/li-et-al-2024-dual-stressors-of-infection-and-warming-can-destabilize-host-microbiomes.pdf\"\n",
    "def extract_abstract(pdf_path):\n",
    "    with open(pdf_path, \"rb\") as file:\n",
    "        reader = PdfReader(file)\n",
    "        first_page_text = reader.pages[0].extract_text()  # Assuming abstract is on the first page\n",
    "\n",
    "        # Split the text into paragraphs (split by double newline as a simple example)\n",
    "        paragraphs = first_page_text.split('\\n\\n')\n",
    "\n",
    "        # Attempt to identify the abstract\n",
    "        for i, paragraph in enumerate(paragraphs):\n",
    "            if 100 < len(paragraph) < 500:  # Assuming abstract length is within this range\n",
    "                print(f\"Possible Abstract Found: {paragraph}\\n\")\n",
    "                return paragraph  # Return the first paragraph that matches the criteria\n",
    "\n",
    "        return \"Abstract not found.\"\n",
    "\n",
    "# Example usage\n",
    "pdf_path = 'path_to_your_pdf.pdf'  # Replace with your PDF file path\n",
    "abstract = extract_class(filepath)\n",
    "print(\"Extracted Abstract:\")\n",
    "print(abstract)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
